<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<title>FAIR for busy biologists: All in One View</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<link rel="stylesheet" type="text/css" href="../assets/styles.css">
<script src="../assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="manifest" href="../site.webmanifest">
<link rel="mask-icon" href="../safari-pinned-tab.svg" color="#5bbad5">
<meta name="msapplication-TileColor" content="#da532c">
<meta name="theme-color" content="#ffffff">
</head>
<body>
    <header id="top" class="navbar navbar-expand-md navbar-light bg-white top-nav incubator"><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-6">
      <div class="large-logo">
        <img alt="Carpentries Incubator" src="../assets/images/incubator-logo.svg"><abbr class="badge badge-light" title="This lesson is in the alpha phase, which means that it has been taught once and lesson authors are iterating on feedback." style="background-color: #FFC700; border-radius: 5px">
          <a href="https://cdh.carpentries.org/the-lesson-life-cycle.html#field-testing-alpha-stage" class="external-link alert-link" style="color: #383838">
            <i aria-hidden="true" class="icon" data-feather="alert-triangle" style="border-radius: 5px"></i>
            Alpha
          </a>
          <span class="visually-hidden">This lesson is in the alpha phase, which means that it has been taught once and lesson authors are iterating on feedback.</span>
        </abbr>
        
      </div>
    </div>
    <div class="selector-container">
      
      
      <div class="dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1">
<li><button class="dropdown-item" type="button" onclick="window.location.href='../aio.html';">Learner View</button></li>
        </ul>
</div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl navbar-light bg-white bottom-nav incubator" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="Carpentries Incubator" src="../assets/images/incubator-logo-sm.svg">
</div>
    <div class="lesson-title-md">
      FAIR for busy biologists
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="search button" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0">
<li class="nav-item">
          <span class="lesson-title">
            FAIR for busy biologists
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/instructor-notes.html">Instructor Notes</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/images.html">Extract All Images</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown">
<hr>
<li><a class="dropdown-item" href="reference.html">Reference</a></li>
          </ul>
</li>
      </ul>
</div>
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
<input class="form-control me-2 searchbox" type="search" placeholder="Search" aria-label="Search"><button class="btn btn-outline-success tablet-search-button" type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="search button"></i>
        </button>
      </fieldset>
</form>
  </div>
<!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  FAIR for busy biologists
</div>

<aside class="col-md-12 lesson-progress"><div style="width: %" class="percentage">
    %
  </div>
  <div class="progress incubator">
    <div class="progress-bar incubator" role="progressbar" style="width: %" aria-valuenow="" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="../aio.html">Learner View</a>
                      </div>
                    </div>
                  </div>
<!--/div.accordion-item-->
                </div>
<!--/div.accordion-flush-->
              </div>
<!--div.sidenav-view-selector -->
            </div>
<!--/div.col -->
      
            <hr>
</div>
<!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Schedule</a>
    </div>
<!--/div.accordion-header-->
        
  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="01-OS-and-FAIR.html">1. Introduction to Open Science and FAIR principles</a>
    </div>
<!--/div.accordion-header-->
        
  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="02-Intro-to-metadata.html">2. Introduction to metadata</a>
    </div>
<!--/div.accordion-header-->
        
  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="03-tidy-data-tables.html">3. Tidy (meta)data tables</a>
    </div>
<!--/div.accordion-header-->
        
  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="04-files-organization.html">4. Working with files</a>
    </div>
<!--/div.accordion-header-->
        
  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush6">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading6">
        <a href="05-reusable-analysis.html">5. Reusable analysis</a>
    </div>
<!--/div.accordion-header-->
        
  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush7">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading7">
        <a href="06-data-repositories.html">6. Public repositories</a>
    </div>
<!--/div.accordion-header-->
        
  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush8">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading8">
        <a href="07-journey-to-FAIR.html">7. Journey to be FAIR</a>
    </div>
<!--/div.accordion-header-->
        
  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width">
<div class="accordion accordion-flush resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul>
<li>
                        <a href="../instructor/key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="../instructor/instructor-notes.html">Instructor Notes</a>
                      </li>
                      <li>
                        <a href="../instructor/images.html">Extract All Images</a>
                      </li>
                      <hr>
<li><a class="dropdown-item" href="reference.html">Reference</a></li>
                    </ul>
</div>
                </div>
              </div>
            </div>
            <hr class="half-width resources">
<a href="../instructor/aio.html">See all in one page</a>
            

            <hr class="d-none d-sm-block d-md-none">
<div class="d-grid gap-1">
            
            </div>
          </div>
<!-- /div.accordion -->
        </div>
<!-- /div.sidebar-inner -->
      </nav>
</div>
<!-- /div.sidebar -->
  </div>
<!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-extra.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <main id="main-content" class="main-content"><div class="container lesson-content">
        
        
<section id="aio-01-OS-and-FAIR"><p>Content from <a href="01-OS-and-FAIR.html">Introduction to Open Science and FAIR principles</a></p>
<hr>
<p>Last updated on 2024-03-22 |
        
        <a href="https://github.com/BioRDM/fair-for-busy-biologists/edit/main/episodes/01-OS-and-FAIR.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 70 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What is Open Science?</li>
<li>How can I benefit from Open Science?</li>
<li>What are the FAIR guidelines?</li>
<li>Why is being FAIR important?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Identify the components of the Open Science movement, their goals,
and motivations.</li>
<li>Explain the key benefits of Open Science.</li>
<li>Recognize the barriers and challenges associated with adopting Open
Science practices.</li>
<li>Identify common issues that hinder data reusability.</li>
<li>Understand the FAIR principles.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>(10 min teaching)</p>
<p>Science thrives on the exchange and development of ideas. The most
efficient scientific progress involves well-informed questions and
experiments, which necessitates the free exchange of data and
information.</p>
<p>All practices that make knowledge and data <strong>freely
available</strong> fall under the umbrella term of <strong>Open
Science/Open Research</strong>. It fosters greater
<strong>reproducibility, transparency, and accessibility</strong> in
science. As science becomes more open, the way we conduct and
communicate scientific findings continuously evolves.</p>
<div id="what-is-open-science" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="what-is-open-science" class="callout-inner">
<h3 class="callout-title">What is Open Science<a class="anchor" aria-label="anchor" href="#what-is-open-science"></a>
</h3>
<div class="callout-content">
<p>Open science is the movement to make scientific research (including
publications, data, physical samples, and software) and its
dissemination <strong>accessible to all levels</strong> of an inquiring
society, amateur or professional.</p>
<p>Open Science represents a new approach to the scientific process
based on <strong>collaborative work</strong> and utilizing
<strong>digital technologies</strong> and new collaborative tools to
disseminate knowledge.</p>
<p>Open science promotes <strong>transparent and accessible
knowledge</strong> that is shared and developed through collaborative
networks.</p>
<p>Characteristics:</p>
<ul>
<li>Utilizing web-based tools to facilitate information exchange and
scientific collaboration</li>
<li>Transparency in experimental methodology, observations, and data
collection</li>
<li>Public availability and reusability of scientific data, methods, and
communications</li>
<li>Diverse outputs and target audiences</li>
</ul>
</div>
</div>
</div>
<div class="section level3">
<h3 id="what-is-the-open-science-movement">What is the Open Science movement?<a class="anchor" aria-label="anchor" href="#what-is-the-open-science-movement"></a>
</h3>
<p>The distribution of knowledge has always had room for improvement.
While the internet was initially developed for military purposes, it was
ultimately used for communication between scientists, providing a viable
path to transform how science is disseminated.</p>
<p>The momentum has grown alongside a shift in how science is
communicated, reflecting the needs of research communities. Open Science
addresses many of the pressing issues we face today, such as impact
factors, data reusability, the reproducibility crisis, and trust in the
public science sector.</p>
<p>Open Science is the movement to increase transparency and
reproducibility of research through the adoption of open best
practices.</p>
<figure><img src=".././fig/OpenScienceBuildingBlocks.jpg" alt="Figure 1. Open Science Building Blocks" class="figure mx-auto d-block"><div class="figcaption">Figure 1. Open Science Building Blocks</div>
</figure><p><em>After <a href="https://www.fosteropenscience.eu/content/what-open-science-introduction" class="external-link">Gema
Bueno de la Fuente</a></em></p>
</div>
<div class="section level3">
<h3 id="open-science-building-blocks">Open Science Building Blocks<a class="anchor" aria-label="anchor" href="#open-science-building-blocks"></a>
</h3>
<ul>
<li><p><strong>Open Access:</strong> Research outputs are hosted in a
way that makes them accessible to everyone. Traditionally, Open Access
referred to journal articles, but now encompasses books, chapters, or
images.</p></li>
<li><p><strong>Open Data:</strong> Data is freely and readily available
to access, reuse, and share. Smaller data sets were often included as
supplemental materials alongside articles in journals. However, they
should be hosted on dedicated platforms for more convenient and improved
access.</p></li>
<li><p><strong>Open Software:</strong> Software with readily available
source code; others can freely use, modify, and share it. Examples
include the coding language and supporting software R and RStudio, as
well as image analysis software like Fiji/ImageJ.</p></li>
<li><p><strong>Open Notebooks:</strong> Lab notebooks hosted online,
readily accessible to all. These are popular among some large funding
bodies and allow anyone to comment on any stage of the experimental
record.</p></li>
<li><p><strong>Open Peer Review:</strong> A system where peer review
reports are published alongside the research work. This can include
reviewers’ reports, correspondence between involved parties, rebuttals,
editorial decisions, etc.</p></li>
<li><p><strong>Citizen Science:</strong> Lay people contribute to
scientific research, most commonly in data collection or image analysis.
Platforms like <a href="https://www.zooniverse.org/" class="external-link">https://www.zooniverse.org/</a>
connect projects with interested laypeople who can actively participate
in research, helping to generate and/or process data that would be
unachievable by a single person.</p></li>
<li><p><strong>Scientific social networks:</strong> Networks of
researchers, often meeting locally in teams but also connected online,
foster open discussions on scientific issues. Many researchers use
traditional social media platforms for this purpose, such as Twitter,
Instagram, various subreddits, discussion channels on Slack/Discord,
etc. There are also dedicated spaces like <a href="https://www.researchgate.net/" class="external-link">https://www.researchgate.net/</a>.</p></li>
<li><p><strong>Open Educational Resources:</strong> Educational
materials that are free for anyone to access and use for learning. These
can be anything from talks, instructional videos. This very course is a
perfect example of that!</p></li>
</ul>
</div>
<div class="section level3">
<h3 id="benefits-of-open-science">Benefits of Open Science<a class="anchor" aria-label="anchor" href="#benefits-of-open-science"></a>
</h3>
<p>Possible benefits and consequences for each Open Science module:</p>
<p><strong>Open Access</strong></p>
<ul>
<li>Faster dissemination of knowledge</li>
<li>Levels the playing field for underfunded institutions that wouldn’t
otherwise be able to access research behind paywalls</li>
<li>Prevents articles from being paid for “three times” (first to
produce, second to publish, third to access) by institutions</li>
<li>Increases access to work by others, leading to greater exposure and
citations</li>
<li>Provides access to research for laypeople, thus increasing public
awareness of science</li>
</ul>
<p><strong>Open Data</strong></p>
<ul>
<li>Ensures data isn’t lost over time - promotes reusability</li>
<li>Accelerates scientific discovery</li>
<li>Provides value for money and reduces redundancy</li>
<li>Allows for statistical re-analysis of data to validate findings</li>
<li>Gives access to datasets not published in papers (e.g., negative
results, large screening data sets)</li>
<li>Provides a way to generate new hypotheses</li>
<li>Enables the combination of multiple data sources to address
questions, providing greater power than a single data source</li>
</ul>
<p><strong>Open Software</strong></p>
<ul>
<li>Excellent resource for learning programming skills</li>
<li>The ability to modify software fosters a supportive community of
users and rapid innovation</li>
<li>Saves time</li>
<li>Enables faster bug fixes</li>
<li>Encourages better error scrutiny</li>
<li>Using the same software/code allows for better reproducibility
between experiments</li>
<li>Requires funding to maintain and update software</li>
</ul>
<p><strong>Open Notebooks</strong></p>
<ul>
<li>Promotes 100% transparent science, allowing input from others at
early stages of experiments</li>
<li>Provides a source for learning about the scientific process</li>
<li>Allows access to experiments and data that otherwise might never get
published</li>
<li>Provides access to ‘negative’ results and failed experiments</li>
<li>Anyone, anywhere in the world, can access projects at any time,
enabling simultaneous input from many users</li>
<li>Offers the possibility of immediate feedback</li>
<li>Provides thorough evidence of the originality of ideas and
experiments, reducing the impact of “scooping”</li>
</ul>
<p><strong>Open Peer Review</strong></p>
<ul>
<li>Visibility leads to more constructive reviews</li>
<li>Mitigates against editorial conflicts of interest and/or biases</li>
<li>Mitigates against reviewer conflicts of interest and/or biases</li>
<li>Allows readers to learn and benefit from reviewers’ comments</li>
</ul>
<p><strong>Open Educational Materials</strong></p>
<ul>
<li>Fosters collaboration between educators and others</li>
<li>Clearly demonstrates how methods are taught (e.g., Carpentries
materials), which can be reproduced anywhere, anytime</li>
<li>Protects materials from becoming technologically obsolete</li>
<li>Authors who prepare or contribute materials can receive credit
(e.g., GitHub)</li>
<li>Enables the reuse of animations and excellent materials (why
reinvent the wheel?)</li>
</ul>
</div>
<div class="section level3">
<h3 id="motivation-money-8-min-teaching">Motivation: Money (8 min teaching)<a class="anchor" aria-label="anchor" href="#motivation-money-8-min-teaching"></a>
</h3>
<p>We must consider the ethical implications that accompany the research
and publication process. Charities and taxpayers fund research, and then
pay again to access the research they already funded.</p>
<p>From an economic viewpoint, scientific outputs generated by public
research are a public good that everyone should be able to use at no
cost.</p>
<p>According to an EU report titled “Cost-benefit analysis for FAIR
research data: <a href="https://op.europa.eu/en/publication-detail/-/publication/d375368c-1a0a-11e9-8d04-01aa75ed71a1" class="external-link">https://op.europa.eu/en/publication-detail/-/publication/d375368c-1a0a-11e9-8d04-01aa75ed71a1</a>”,
€10.2 billion is lost every year due to inaccessible data (with an
additional €16 billion lost if we consider data reuse and research
quality).</p>
<p>Open Science aims to make research and research data accessible to
those who funded the research, such as charities and taxpayers.</p>
<p>The majority of large funding bodies in the UK and other countries
are now making Open Access publication a condition of funding. As a
result, Open Access is becoming the most widespread aspect of the Open
Science movement, adopted by the majority of researchers.</p>
</div>
<div class="section level3">
<h3 id="personal-motivators">Personal motivators<a class="anchor" aria-label="anchor" href="#personal-motivators"></a>
</h3>
<p>Open Science offers advantages to many stakeholders in science
(including the research community, funding bodies, the public, and even
journals), leading to a push for widespread adoption of Open Science
practices.</p>
<p>Large UK funding bodies like The Wellcome Trust are strong supporters
of Open Science. The example of Open Access demonstrates that
enforcement by funders (<em>the stick</em>) can lead to widespread
adoption. But what about the personal motivators, <em>the
carrots</em>?</p>
<div id="exercise-1-personal-benefits-of-being-open-5-min" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="exercise-1-personal-benefits-of-being-open-5-min" class="callout-inner">
<h3 class="callout-title">Exercise 1: Personal benefits of being “open”
(5 min)<a class="anchor" aria-label="anchor" href="#exercise-1-personal-benefits-of-being-open-5-min"></a>
</h3>
<div class="callout-content">
<p>Open Science practices offer a variety of advantages for researchers.
Read through the list below and consider which benefits resonate most
with you.</p>
<p>Select two the most important/attractive for you and mark them with
+1, select the two least important for you and mark them with 0</p>
<ul>
<li>receive higher citations</li>
<li>complying with funders’ policies</li>
<li>get extra value from your work (e.g. collaborators, reuse by
modellers, ML specialists)</li>
<li>demonstrate research impact</li>
<li>save own time (reproducibility but also communication overhead)</li>
<li>become pioneers</li>
<li>distinguish yourself from the crowd</li>
<li>plan successful research proposals</li>
<li>gain valuable experience</li>
<li>form community</li>
<li>increased speed and/or ease of writing papers</li>
<li>speed up and help with peer review</li>
<li>build reputation and presence in the science community</li>
<li>evidence of your scientific rigour and work ethic</li>
<li>avoid embarrassment/disaster when you cannot reproduce your
results</li>
</ul>
<p>Can you think of other benefits?</p>
<p>How personal benefits of Open Science compare to the benefits for the
(scientific) society?</p>
</div>
</div>
</div>
<p>(5 min teaching)</p>
<p>Open Science offers distinct advantages for both researchers and
society at large. The public benefits from immediate access to research
outputs, leading to faster scientific progress and innovation.</p>
<p>For researchers, the benefits take longer time, as open data and
publications need to time to lead to citations, collaborations, and
recognition within the scientific community</p>
</div>
<div class="section level3">
<h3 id="dora-declaration-on-research-assessment">DORA: Declaration on Research Assessment<a class="anchor" aria-label="anchor" href="#dora-declaration-on-research-assessment"></a>
</h3>
<p>The San Francisco Declaration on Research Assessment (DORA)
emphasizes the limitations of using metrics like Journal Impact Factors
(JIF) to solely evaluate research. DORA advocates for assessing research
based on its inherent merit and contributions, promoting fairer and more
transparent evaluation practices. This shift acknowledges the importance
of research quality, openness, and its broader societal impact.</p>
</div>
<div class="section level3">
<h3 id="funders-embrace-dora-principles">Funders Embrace DORA Principles<a class="anchor" aria-label="anchor" href="#funders-embrace-dora-principles"></a>
</h3>
<p>Research funders worldwide are increasingly endorsing DORA
principles. Leading institutions like Wellcome Trust and Cancer Research
UK incorporate DORA criteria into their funding applications. These
funders prioritize research outputs, mentorship contributions, and
public engagement plans, supporting research that generates valuable
knowledge, fosters collaboration, and benefits society.</p>
</div>
<div class="section level3">
<h3 id="narrative-cv-as-a-dora-compliant-assessment-tool">Narrative CV as a DORA-Compliant Assessment Tool:<a class="anchor" aria-label="anchor" href="#narrative-cv-as-a-dora-compliant-assessment-tool"></a>
</h3>
<p>The Narrative CV aligns with DORA principles by focusing on key
dimensions that reflect a researcher’s contributions:</p>
<ul>
<li>Generation of Knowledge: Acknowledging diverse outputs such as
datasets, patents, and software.</li>
<li>Development of Individuals and Collaborations: Highlighting
mentorship and collaborative endeavors that enrich the research
ecosystem.</li>
<li>Societal and Economic Impact: Demonstrating the societal and
economic impacts of research beyond academic circles.</li>
<li>Supporting the Research Community: Engaging in open science
practices and ensuring the accessibility of research outputs.</li>
</ul>
<p>This framework prioritizes open science practices to maximize
research impact and visibility. Additionally, new metrics like retweets,
online views/downloads, discussions, and media coverage are considered,
providing a more comprehensive understanding of research impact in the
digital age.</p>
<p><strong>Why Embrace Open Practices?</strong></p>
<p>Open Science practices not only uphold ethical research conduct but
also enhance the credibility and reach of your work. The Narrative CV
and the adoption of DORA by leading funders exemplify the research
community’s shift towards a more transparent and equitable assessment
paradigm. Choosing open practices demonstrates not just integrity, but
also a recognition that authenticity and transparency are fundamental to
scientific progress. After all, timestamps and meticulous documentation
make faking open practices far more difficult than simply adhering to
them.</p>
</div>
<div class="section level3">
<h3 id="barriers-and-risks-of-the-open-science-movement">Barriers and risks of the Open Science movement:<a class="anchor" aria-label="anchor" href="#barriers-and-risks-of-the-open-science-movement"></a>
</h3>
<div id="exercise-2-why-we-are-not-doing-open-science-already-about-5-min" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-2-why-we-are-not-doing-open-science-already-about-5-min" class="callout-inner">
<h3 class="callout-title">Exercise 2: Why we are not doing Open Science
already (about 5 min)<a class="anchor" aria-label="anchor" href="#exercise-2-why-we-are-not-doing-open-science-already-about-5-min"></a>
</h3>
<div class="callout-content">
<p>Discuss the challenges and potential drawbacks associated with Open
Science practices.</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1">Show me the solution</h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<ul>
<li>
<strong>Data Sensitivity:</strong> Anonymizing data from certain
sources, like administrative health records, can be complex and
time-consuming.</li>
<li>
<strong>Intellectual Property (IP) Concerns:</strong> Researchers
might hesitate to share data if it could compromise potential patents or
other intellectual property.</li>
<li>
<strong>Misuse of Data:</strong> Open data carries a risk of
misinterpretation or misuse, potentially leading to the spread of
misinformation (e.g., “fake news”).</li>
<li>
<strong>Lack of Recognition for Negative Results:</strong>
Publishing negative or inconclusive findings can be less rewarding in
the current academic evaluation system.</li>
<li>
<strong>Time and Cost:</strong> Making research outputs open and
user-friendly requires time and resources for proper data curation,
storage, and dissemination. This can be especially challenging for large
datasets.</li>
<li>
<strong>Lack of Expertise:</strong> Researchers might not have the
necessary skills or training to effectively manage and share data
openly.</li>
<li>
<strong>Fear of Criticism:</strong> The prospect of open peer review
or public scrutiny can be daunting, leading some researchers to shy away
from open science practices.</li>
</ul>
</div>
</div>
</div>
</div>
<p>(8 min teaching)</p>
<p>There are valid reasons why some researchers hesitate to embrace Open
Science entirely.</p>
<p><strong>Data sensitivity</strong> is a major concern. Data privacy
laws require careful handling of sensitive data, and anonymization can
be a complex process.</p>
<p>Anonymising data to desensitise it can help overcome this
barrier.</p>
<p><strong>Intellectual property (IP) considerations</strong> are
another hurdle. Researchers may be hesitant to share data that could
compromise the patenting or commercialization of their discoveries.
However, careful data filtering can often address these concerns. For IP
protection it is the timeline of public disclosure that matters not fact
making them public.</p>
<p><strong>Pre-prints</strong>, while valuable for rapid knowledge
dissemination, can be misused, particularly in fast-moving fields like
public health research. Unreviewed pre-prints have the potential to
mislead policymakers and the public if not interpreted with caution.
This can result in political and health decision making based on faulty
data, which is counter to societies’ best interest.</p>
<p>The fear of negative feedback can also be a barrier. However, open
peer review is a crucial part of the scientific process. Identifying and
correcting errors through open review strengthens research and fosters a
culture of transparency.</p>
<p>We should seek for our work to be scrutinized and errors to be
pointed out, and is the sign of a competent scientist. One should rather
have errors pointed out rather than risking that irreproducible data
might cause even more embarrassment and disaster.</p>
<p>Lack of Expertise: Researchers might not have the necessary skills or
training to effectively manage and share data openly.</p>
<p>One of <strong>the biggest barriers are the costs</strong> involved
in “being Open”. The <strong>time and financial investment</strong>
required for Open Science practices can be a significant barrier. Making
data readily accessible and usable takes effort, and data storage can be
expensive, especially for large datasets.</p>
<p>For example, microscopy datasets reach sizes in terabytes, making
such data accessible for 10 years involves serious financial
commitment.</p>
</div>
<section id="being-fair"><h2 class="section-heading">Being FAIR<a class="anchor" aria-label="anchor" href="#being-fair"></a>
</h2>
<hr class="half-width">
<p>We’ve explored the advantages of Open Science practices for both the
scientific community and individual researchers. While Open Access has
made recent biomedical publications readily available, the same
accessibility often isn’t the case for the underlying data and
software.</p>
<div id="what-is-data" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="what-is-data" class="callout-inner">
<h3 class="callout-title">What is Data?<a class="anchor" aria-label="anchor" href="#what-is-data"></a>
</h3>
<div class="callout-content">
<p>The term “scientific data” encompasses a wider range than many might
initially think. It’s not limited to just numbers in spreadsheets! Data
can include:</p>
<ul>
<li>
<strong>Images:</strong> Microscopy images, but also gels, blots,
and other visual representations of findings.</li>
<li>
<strong>Biological Information:</strong> Details about research
materials, like specific strains, cell lines, or patient
demographics.</li>
<li>
<strong>Biological Models:</strong> Computational models used in
simulations or analyses.</li>
<li>
<strong>Protocols:</strong> Step-by-step procedures for lab
experiments or data collection methods.</li>
<li>
<strong>Code:</strong> Scripts, analysis routines, and custom
software used to generate results.</li>
</ul>
<p>While there are specific best practices for sharing code, it’s still
considered a form of research data.</p>
</div>
</div>
</div>
<p>Let’s delve into the challenges associated with accessing and using
data from published biological research.</p>
<div id="exercise-3-impossible-protocol" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-3-impossible-protocol" class="callout-inner">
<h3 class="callout-title">Exercise 3: Impossible protocol<a class="anchor" aria-label="anchor" href="#exercise-3-impossible-protocol"></a>
</h3>
<div class="callout-content">
<p>(5 min breakout, plus 10 talking about both problems)</p>
<p>You need to do a western blot to identify Titin proteins, the largest
proteins in the body, with a molecular weight of 3,800 kDa. You found an
antibody sold by Sigma Aldrich that has been validated in western blots
and immunofluorescence. Sigma Aldrich lists the <a href="https://doi.org/10.1002/acn3.50831" class="external-link">Yu et al., 2019</a> paper as
reference.</p>
<p>Find details of how to separate and transfer this large protein in
the reference paper.</p>
<ul>
<li>Hint 1: Methods section has a Western blot analysis
subsection.<br>
</li>
<li>Hint 2: Follow the references.</li>
</ul>
<p>Would you say that the methods was Findable? Accessible?
Reusable?</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2">Show me the solution</h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" data-bs-parent="#accordionSolution2" aria-labelledby="headingSolution2">
<div class="accordion-body">
<ul>
<li>Ref 17 will lead you to <a href="https://doi.org/10.1002/ana.24102" class="external-link">this paper</a>, which first of
all is not Open Access</li>
<li>Access the paper through your institutions (if you can) and find the
‘Western Blotting’ protocol on page 232 which will show the following
(Screenshot from the methods section from <a href="https://doi.org/10.1002/ana.24102" class="external-link">Evilä et al 2014</a>):</li>
<li><figure><img src=".././fig/impossible_protocol.png" alt="Figure 1. Impossible Protocol" class="figure mx-auto d-block"><div class="figcaption">Figure 1. Impossible Protocol</div>
</figure></li>
<li>“Western blotting were performed according to standard methods.” -
with no further reference to these standard methods, describing these
methods, or supplementary material detailing these methods</li>
<li>This methodology is unfortunately a true dead end and we thus can’t
easily continue our experiments!</li>
</ul>
</div>
</div>
</div>
</div>
<div id="impossible-numbers" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="impossible-numbers" class="callout-inner">
<h3 class="callout-title">Impossible numbers<a class="anchor" aria-label="anchor" href="#impossible-numbers"></a>
</h3>
<div class="callout-content">
<p><a href="https://doi.org/10.1093/jxb/err244" class="external-link">Ikram 2014</a> paper
contains data about various metabolites in different accessions
(genotypes) of Arabidopsis plant. You would like to calculate average
nitrogen content in plants grown under normal and nitrogen limited
conditions. Please calculate the average (over genotypes) nitrogen
content for the two experimental conditions.</p>
<ul>
<li>Hint 1. Data are in Supplementary data<br>
</li>
<li>Hint 2. Search for nitrogen in paper text to identify the correct
data column.</li>
</ul>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3">Show me the solution</h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" data-bs-parent="#accordionSolution3" aria-labelledby="headingSolution3">
<div class="accordion-body">
<ul>
<li>Finding the right table and column containing the relevant data is
already problematic as the headers are obscured so they need to decoded
using manuscript</li>
<li>Data in pdf table so they cannot be readily used in
calculations</li>
<li>Depending on the software used to open (and the way the pdf was
created), the local machine international settings, copying the data
into Excel can bring unexpected results <img src=".././fig/03-average_to_excel1.png" alt="Figure 2. Pdf data copied to Excel" class="figure"><br><em>Data needs parsing after coping to Excel</em> <img src=".././fig/03-average_to_excel1.png" alt="Figure 2. The same data copied to Excel with polish locale" class="figure"><br><em>The same data copied to Excel with polish locale has been converted
to dates</em>
</li>
<li>In general pdf tables cannot be read programmatically from R or
Python.</li>
</ul>
</div>
</div>
</div>
</div>
<p><strong>The above examples illustrate the typical challenges in
accessing research data and software.</strong></p>
<p>Firstly, data, protocols, and software often lack a distinct
identity, existing only as supplements to publications.</p>
<p>Second, accessibility and reusability are often compromised. For
instance, all details might be buried within a single supporting
information PDF file. These files often contain “printed” numerical
tables or even source code, both of which require manual re-entry for
use by others. Additionally, data may be shared in proprietary file
formats specific to a particular vendor, inaccessible without the
accompanying software. Finally, data files are often provided without
detailed descriptions beyond the full article text, hindering
reusability.</p>
<p>In our examples, the protocol was difficult to <strong>find</strong>
(the loops), <strong>access</strong> (paywall), and
<strong>reuse</strong> due to a lack of necessary details (dead-end).
Similarly, in the second example, the data were not
<strong>interoperable</strong> or <strong>reusable</strong> as they were
only available as a figure graph.</p>
<p>To address these problems, the FAIR principles were designed.</p>
<p>In our examples, the protocol was difficult to <strong>find</strong>
(the loops), difficult to <strong>access</strong> (pay wall), and not
<strong>reusable</strong> as it lacked the necessary details
(dead-end).</p>
<p>In the second example the data were not
<strong>interoperable</strong> and <strong>reusable</strong> as their
were only available as a figure graph.</p>
<p>To avoid such problems FAIR principles were designed.</p>
<p><img src=".././fig/FAIR.png" alt="Figure 2. FAIR principles" class="figure"> After <a href="https://commons.wikimedia.org/wiki/File:FAIR_data_principles.jpg" class="external-link">SangyaPundir</a></p>
<p>(10 min teaching)</p>
<div id="fair-principles" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="fair-principles" class="callout-inner">
<h3 class="callout-title">FAIR Principles<a class="anchor" aria-label="anchor" href="#fair-principles"></a>
</h3>
<div class="callout-content">
<p>In 2016, the FAIR Guiding Principles for scientific data management
and stewardship: <a href="https://www.nature.com/articles/sdata201618" class="external-link">https://www.nature.com/articles/sdata201618</a>
were published in Scientific Data. The original guideline emphasized
“machine-actionability” - the ability of computer systems to
automatically process data. However, the focus has shifted towards
human-centric accessibility, acknowledging the limitations of
user-friendly tools for standardized metadata.</p>
<p><strong>Findable:</strong> Data and metadata should be easily
discoverable by both humans and computers. Standardized and persistent
identifiers (PIDs) and machine-readable metadata are crucial for
automatic dataset and service discovery.</p>
<p><strong>Accessible:</strong> (Meta)data should be retrievable using a
standardized and open communication protocol (including authentication
and authorization) based on their identifier. Even if the data itself
becomes unavailable, the metadata should remain accessible.</p>
<p><strong>Interoperable:</strong> Data should be combinable and usable
with other data or tools. Open and interpretable data formats are
essential for various tools. Both data and metadata should utilize
vocabularies that follow FAIR principles to promote
interoperability.</p>
<p><strong>Reusable:</strong> FAIR aims to optimize data reuse.
Well-described metadata and data facilitate replication and integration
in different contexts. Clear and accessible licenses should be provided
to govern (meta)data reuse.</p>
</div>
</div>
</div>
</section><section id="fair-in-biological-practice"><h2 class="section-heading">FAIR in biological practice<a class="anchor" aria-label="anchor" href="#fair-in-biological-practice"></a>
</h2>
<hr class="half-width">
<div class="section level4">
<h4 id="findable-accessible">Findable &amp; Accessible<a class="anchor" aria-label="anchor" href="#findable-accessible"></a>
</h4>
<p>Deposit your data in a reputable external public repository.</p>
<p>These repositories provide persistent identifiers (PIDs) for easy
discovery, along with options for cataloging, advanced metadata
searching, and download statistics. Some repositories can even host
private data or offer embargo periods to delay full data access.</p>
<p><strong>General “data agnostic” repositories include:</strong></p>
<ul>
<li>
<a href="http://datadryad.org" class="external-link">Dryad</a>,</li>
<li>
<a href="http://zenodo.org" class="external-link">Zenodo</a>,</li>
<li>
<a href="http://figshare.com" class="external-link">FigShare</a>,</li>
<li>
<a href="http://thedata.org" class="external-link">Dataverse</a>.</li>
</ul>
<p>*Domain-specific examples include:**</p>
<ul>
<li>
<a href="https://www.uniprot.org/" class="external-link">UniProt</a> protein data,</li>
<li>
<a href="https://www.ncbi.nlm.nih.gov/genbank/" class="external-link">GenBank</a> sequence
data,</li>
<li>
<a href="https://www.ebi.ac.uk/metabolights/" class="external-link">MetaboLights</a>
metabolomics data</li>
<li>
<a href="https://github.com/" class="external-link">GitHub</a> for code.</li>
</ul>
<p><em>We will cover repositories in more details in a later
episode.</em></p>
<div id="accordionSpoiler1" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler1" aria-expanded="false" aria-controls="collapseSpoiler1">
  <h3 class="accordion-header" id="headingSpoiler1">
<div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>What are persistent identifiers (PIDs)</h3>
</button>
<div id="collapseSpoiler1" class="accordion-collapse collapse" data-bs-parent="#accordionSpoiler1" aria-labelledby="headingSpoiler1">
<div class="accordion-body">
<p>A persistent identifier (PID) is a long-lasting reference to a
digital resource. It typically consists of two parts:</p>
<ul>
<li>A service that locates the resource over time, even if its location
changes.</li>
<li>A unique identifier that distinguishes the resource or concept from
others.</li>
</ul>
<p>PIDs address the problem of accessing cited resources, particularly
in academic literature, where web addresses (links) often change over
time, leading to broken links.</p>
<p>Several services and technologies (schemes) provide PIDs for various
objects (digital, physical, or abstract). One of the most common is the
Digital Object Identifier (DOI) (<a href="https://www.doi.org/" class="external-link">https://www.doi.org/</a>), recognizable by
the prefix “doi.org” in web links. For instance, this link (<a href="https://pubmed.ncbi.nlm.nih.gov/26978244/" class="external-link">https://pubmed.ncbi.nlm.nih.gov/26978244/</a>)
resolves to the paper that describes FAIR principles.</p>
<p>Public repositories often maintain stable web addresses for their
content, following the convention
<code>http://repository.address/identifier</code>. These are often
called permalinks. For well-established services, permalinks can be
considered PIDs.</p>
<p>For example, this link <a href="http://identifiers.org/SO:0000167" class="external-link">http://identifiers.org/SO:0000167</a>
points to a page defining the promoter role and can be used to annotate
a DNA sequence performing such a role during transcription.</p>
</div>
</div>
</div>
</div>
</div>
<div class="section level4">
<h4 id="interoperable">Interoperable<a class="anchor" aria-label="anchor" href="#interoperable"></a>
</h4>
<ul>
<li>Use common and ideally free file formats (domain-specific options
might exist).</li>
<li>Always use <code>.csv</code> or <code>.xls</code> for numerical
data. <strong>Never</strong> share data tables as Word or PDF
documents.</li>
<li>Provide the underlying numerical data for all plots and graphs.</li>
<li>Convert proprietary binary formats to open formats. For example,
convert Snapgene to GenBank, or microscopy multistack images to
OME-TIFF.</li>
</ul>
</div>
<div class="section level4">
<h4 id="reusable">Reusable<a class="anchor" aria-label="anchor" href="#reusable"></a>
</h4>
<p>Describe your data well with comprehensive metadata.</p>
<ul>
<li>Write a README file that describes your data.</li>
<li>Use descriptive column headers in your data tables.</li>
<li>Organize your data tables for analysis-friendliness (tidy data
principles).</li>
<li>Provide as much detail as possible through rich metadata.</li>
<li>Utilize appropriate metadata formats (e.g., SBML, SBOL).</li>
<li>Adhere to Minimum Information Standards (MIS) for your field.</li>
</ul>
<p>Describing data in sufficient detail is often the most challenging
aspect of data sharing. We’ll cover this in more detail later.</p>
<ol start="2" style="list-style-type: decimal">
<li>Include license files.</li>
</ol>
<p>Licenses explicitly state the conditions and terms under which your
data and software can be reused. Here are some recommendations:</p>
<ul>
<li>For data, we recommend the <a href="https://creativecommons.org/licenses/by/4.0/" class="external-link">Creative Commons
Attribution (CC BY)</a> license.</li>
<li>For code, consider a permissive open-source license such as <a href="https://opensource.org/licenses/MIT" class="external-link">MIT</a>, <a href="https://opensource.org/licenses/BSD-2-Clause" class="external-link">BSD</a>, or Apache
<a href="http://www.apache.org/licenses/" class="external-link">Apache license</a>
licenses.</li>
</ul>
<div id="copyright-and-data" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="copyright-and-data" class="callout-inner">
<h3 class="callout-title">Copyright and data<a class="anchor" aria-label="anchor" href="#copyright-and-data"></a>
</h3>
<div class="callout-content">
<p>Software code (the text itself) automatically receives default
copyright protection, preventing others from copying or modifying it.
Only by adding an explicit license can you permit others to reuse
it.</p>
<p>Data, being factual, cannot be copyrighted. <strong>So why, do we
need a license?</strong></p>
<p>While the data itself cannot be copyrighted, the way it’s presented
can be. The extent to which it’s protected ultimately needs to be
settled by a court.</p>
<p>“Good actors” will refrain from using your data to avoid legal risks.
However, “bad actors” might ignore the risk or have the resources to
fight legal battles.</p>
</div>
</div>
</div>
<div id="exercise-4-example-of-fair-data" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-4-example-of-fair-data" class="callout-inner">
<h3 class="callout-title">Exercise 4: Example of FAIR data<a class="anchor" aria-label="anchor" href="#exercise-4-example-of-fair-data"></a>
</h3>
<div class="callout-content">
<p>(5 min breakout, plus 3 min showing answers)</p>
<p>Zenodo is general data repository. Have a look at the dataset record
with COVID-19 data: <a href="https://doi.org/10.5281/zenodo.6339631" class="external-link">https://doi.org/10.5281/zenodo.6339631</a></p>
<p>Identify how each of F.A.I.R principles has been met.<br><em>Hint: navigate to linked github record to easily access the README
file</em></p>
</div>
</div>
</div>
<div id="accordionSolution4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution4" aria-expanded="false" aria-controls="collapseSolution4">
  <h4 class="accordion-header" id="headingSolution4">Show me the solution</h4>
</button>
<div id="collapseSolution4" class="accordion-collapse collapse" data-bs-parent="#accordionSolution4" aria-labelledby="headingSolution4">
<div class="accordion-body">
<ul>
<li>F: The dataset is identified by a PID (doi). It can be found by its
ID. It human accessible description and keywords, both suitable for
discovery.</li>
<li>A: Data can be downloaded using standard browser.</li>
<li>I: Dataset entries are in common formats: csv, R, jpg</li>
<li>I: Dataset is linked to publication, github record and project
website</li>
<li>R: The record contains rich metadata in README file, including files
structure and the detailed tables formats.</li>
<li>R: Data are released under open Creative Commons Attribution
Licence</li>
</ul>
</div>
</div>
</div>
</div>
<p>The FAIR acronym is sometimes accompanied with the following labels:
* Findable - Citable * Accessible - Trackable and countable *
Interoperable - Intelligible * Reusable - Reproducible</p>
<ul>
<li>
<strong>Findable - Citable:</strong> Findable data has a unique
identifier, ensuring proper attribution to the creators.</li>
<li>
<strong>Accessible - Trackable and Countable:</strong> Accessible
data allows for monitoring usage statistics (e.g., downloads, user
locations) to understand its reach and impact.</li>
<li>
<strong>Interoperable - Intelligible:</strong> Interoperable data is
understandable not only by current users but also by future users, even
if they lack access to the specific software used for analysis. This is
achieved through the use of standard formats and clear documentation.
The future you may not remember abreviations and ad-hoc conventions you
used before</li>
<li>
<strong>Reusable - Reproducible:</strong> Well-documented data with
sufficient detail allows for reproducing the experiments, facilitating
verification and building upon existing knowledge. This saves time and
resources.</li>
</ul>
</div>
</section><section id="fair-vs-open-science-2-min-teaching"><h2 class="section-heading">FAIR vs Open Science (2 min teaching)<a class="anchor" aria-label="anchor" href="#fair-vs-open-science-2-min-teaching"></a>
</h2>
<hr class="half-width">
<p><strong>FAIR is not synonymous with Open.</strong> FAIR guidelines
primarily require open access to the metadata record, describing the
data’s existence, a user-friendly PID for reference, and potentially
some restrictions on accessing the actual data files (e.g., login
required).</p>
<p>However, FAIR data is well-positioned for becoming Open data. Since
it’s already accessible online, uses interoperable formats, and comes
with thorough documentation, transitioning to fully open access becomes
a relatively simple step whenever the data owner decides it’s no longer
a risk.</p>
<p>Conversely, Open data lacking FAIR characteristics holds minimal
value. Poorly described data in non-standard formats remains unusable
even if it’s publicly available.</p>
<div id="open-science-and-fair-quiz-5-min" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="open-science-and-fair-quiz-5-min" class="callout-inner">
<h3 class="callout-title">Open Science and FAIR Quiz (5 min)<a class="anchor" aria-label="anchor" href="#open-science-and-fair-quiz-5-min"></a>
</h3>
<div class="callout-content">
<p>Which of the following statements about the OS and FAIR are
true/false?</p>
<ul>
<li>Open Science relies strongly on the Internet</li>
<li>Open Access eliminates publishing costs</li>
<li>Open Data facilitates re-use</li>
<li>Open Data can increases confidence in research findings</li>
<li>In Open Peer Review, readers vote on publication acceptance<br>
</li>
<li>Open Access permits the whole society to benefit from scientific
findings</li>
<li>Citizen Science engages the public in the research process</li>
<li>Release of public datasets is important for career progression</li>
<li>F in FAIR stands for free.</li>
<li>Only figures presenting results of statistical analysis need
underlying numerical data.</li>
<li>Sharing numerical data as a .pdf in Zenodo is FAIR.</li>
<li>Sharing numerical data as an Excel file via Github is not FAIR.</li>
<li>Group website is a good place to share your data.</li>
<li>Data should always be converted to Excel or .csv files in order to
be FAIR.</li>
<li>A DOI of a dataset helps in getting credit.</li>
<li>FAIR data are peer reviewed.</li>
<li>FAIR data accompany a publication.</li>
</ul>
</div>
</div>
</div>
<div id="accordionSolution5" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution5" aria-expanded="false" aria-controls="collapseSolution5">
  <h4 class="accordion-header" id="headingSolution5">Show me the solution</h4>
</button>
<div id="collapseSolution5" class="accordion-collapse collapse" data-bs-parent="#accordionSolution5" aria-labelledby="headingSolution5">
<div class="accordion-body">
<ul>
<li>Open Science relies strongly on the Internet T</li>
<li>Open Access eliminates publishing costs F</li>
<li>Open Data facilitates re-use T</li>
<li>Open Data increases confidence in research findings T</li>
<li>In Open Peer Review, readers vote on publication acceptance F</li>
<li>Open Access permits the whole society to benefit from scientific
findings T</li>
<li>Citizen Science engages the public in the research process T</li>
<li>Release of public datasets is important for career progression
T</li>
<li>F in FAIR stands for free. F</li>
<li>Only figures presenting results of statistical analysis need
underlying numerical data. F</li>
<li>Sharing numerical data as a .pdf in Zenodo is FAIR. F</li>
<li>Sharing numerical data as an Excel file via Github is not FAIR.
F</li>
<li>Group website is a good place to share your data. F</li>
<li>Data should always be converted to Excel or .csv files in order to
be FAIR. F</li>
<li>A DOI of a dataset helps in getting credit. T</li>
<li>FAIR data are peer reviewed. F</li>
<li>FAIR data accompany a publication. F</li>
</ul>
</div>
</div>
</div>
</div>
<div class="section level3">
<h3 id="where-to-next">Where to next<a class="anchor" aria-label="anchor" href="#where-to-next"></a>
</h3>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">SH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode sh" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a> <span class="ex">Further</span> reading/links:</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a> <span class="ex">-</span> [Challenges <span class="kw">&amp;</span> <span class="ex">benefits</span> of OS]<span class="er">(</span><span class="ex">https://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.3000246</span><span class="kw">)</span></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a> <span class="ex">-</span> [Centre for Open Science]<span class="er">(</span><span class="ex">https://www.cos.io/</span><span class="kw">)</span></span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a> <span class="ex">-</span> [Ted talk supporting OS]<span class="er">(</span><span class="ex">https://www.youtube.com/watch?v=c-bemNZ-IqA</span><span class="kw">)</span></span></code></pre>
</div>
</div>
</section><section id="attribution"><h2 class="section-heading">Attribution<a class="anchor" aria-label="anchor" href="#attribution"></a>
</h2>
<hr class="half-width">
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">SH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode sh" tabindex="0"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a> <span class="ex">Content</span> of this episode was adapted from:</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a> <span class="ex">*</span> Wiki [Open Science]<span class="er">(</span><span class="ex">https://en.wikipedia.org/wiki/Open_science</span><span class="kw">)</span></span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a> <span class="ex">*</span> [European Open Science Cloud]<span class="er">(</span><span class="ex">https://www.eosc-hub.eu/open-science-info</span><span class="kw">)</span></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a> <span class="ex">*</span> [Science is necessarily collaborative <span class="at">-</span> The Biochemist article]<span class="er">(</span><span class="ex">https://portlandpress.com/biochemist/article/42/3/58/225220/Science-is-necessarily-collaborative</span><span class="kw">)</span><span class="bu">.</span></span></code></pre>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points<a class="anchor" aria-label="anchor" href="#keypoints1"></a>
</h3>
<div class="callout-content">
<ul>
<li>Open Science increases transparency in research</li>
<li>Publicly funded science should be publicly available</li>
<li>FAIR stands for Findable Accessible Interoperable Reusable</li>
<li>FAIR assures easy reuse of data underlying scientific findings</li>
</ul>
</div>
</div>
</div>
<!-- 
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use. 
 -->
</section></section><section id="aio-02-Intro-to-metadata"><p>Content from <a href="02-Intro-to-metadata.html">Introduction to metadata</a></p>
<hr>
<p>Last updated on 2024-03-22 |
        
        <a href="https://github.com/BioRDM/fair-for-busy-biologists/edit/main/episodes/02-Intro-to-metadata.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 35 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What is metadata?</li>
<li>What do we use metadata for?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Recognize what metadata is</li>
<li>Distinguish different types of metadata</li>
<li>Know how to decide what to include in metadata</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>(13 min teaching)</p>
<section id="what-is-or-are-metadata"><h2 class="section-heading">What is (or are) metadata?<a class="anchor" aria-label="anchor" href="#what-is-or-are-metadata"></a>
</h2>
<hr class="half-width">
<p>Simply put, metadata is data about the data. Sound confusing? Let’s
clarify: metadata is the <em>description of data</em>. It allows a
deeper understanding of data and provides insight for its
interpretation. Hence, your metadata should be considered as important
as your data. Furthermore, metadata plays a very important role in
making your data FAIR. It should be continuously added to your research
data (not just at the beginning or end of a project!). Metadata can be
produced in an automated way (e.g., when you capture a microscopy image,
usually the accompanying software saves metadata as part of it) or
manually.</p>
<ins>
Let’s take a look at an example:
</ins>
<p>This is a confocal microscopy image of a <em>C. elegans</em> nematode
strain used as a proteostasis model (Pretty! Isn’t it?). The image is
part of the raw data associated with <a href="https://doi.org/10.1016/j.celrep.2019.12.078" class="external-link">Goya et al.,
2020</a>, which was deposited in a Public Omero Server<br><a href="https://publicomero.bio.ed.ac.uk/webclient/?show=project-58" class="external-link">Project</a><br><a href="https://publicomero.bio.ed.ac.uk/webclient/?show=dataset-231" class="external-link">Figure1
set</a></p>
<p><img src=".././fig/04-microscopy_with_metadata.png" alt="nematode_confocal_microscopy_image" width="600" class="figure"><em>Figure credits: María Eugenia Goya</em></p>
<p>What information <strong>can you get</strong> from the image, without
the associated description (metadata)?</p>
<p>Let’s see the associated metadata of the image and the dataset to
which it belongs:</p>
<div id="callout1" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout<a class="anchor" aria-label="anchor" href="#callout1"></a>
</h3>
<div class="callout-content">
<p>## Image metadata</p>
<p>Name: OP50 D10Ad_06.czi<br>
Image ID: 3485<br>
Owner: Maria Eugenia Goya<br>
ORCID: 0000-0002-5031-2470</p>
<p>Acquisition Date: 2018-12-12 17:53:55<br>
Import Date: 2020-04-30 22:38:59<br>
Dimensions (XY): 1344 x 1024<br>
Pixels Type: uint16<br>
Pixels Size (XYZ) (µm): 0.16 x 0.16 x 1.00<br>
Z-sections/Timepoints: 56 x 1<br>
Channels: TL DIC, TagYFP<br>
ROI Count: 0</p>
<p>Tags: time course; day 10; adults; food switching; E. coli OP50;
NL5901; C. elegans</p>
<p>## Dataset metadata</p>
<p>Name: Figure2_Figure2B<br>
Dataset ID: 263<br>
Owner: Maria Eugenia Goya<br>
ORCID: 0000-0002-5031-2470</p>
<p>Description:<br>
The datasets contains a time course of α-syn aggregation in NL5901 C.
elegans worms after a food switch at the L4 stage:</p>
<p>E. coli OP50 to OP50<br>
Day 01 adults<br>
Day 03 adults<br>
Day 05 adults<br>
Day 07 adults<br>
Day 10 adults<br>
Day 13 adults</p>
<p>E. coli OP50 to B. subtilis PXN21<br>
Day 01 adults<br>
Day 03 adults<br>
Day 05 adults<br>
Day 07 adults<br>
Day 10 adults<br>
Day 13 adults</p>
<p>Images were taken at 6 developmental timepoints (D1Ad, D3Ad, D5Ad,
D7Ad, D10Ad, D13Ad)</p>
<p>* Some images contain more than one nematode.</p>
<p>Each image contains ~30 (or more) Z-sections, 1 µmeters apart. The
TagYFP channel is used to follow the alpha-synuclein particles. The TL
DIC channel is used to image the whole nematode head.</p>
<p>These images were used to construct Figure 2B of the Cell Reports
paper (<a href="https://doi.org/10.1016/j.celrep.2019.12.078" class="external-link uri">https://doi.org/10.1016/j.celrep.2019.12.078</a>).</p>
<p>Creation date: 2020-04-30 22:16:39</p>
<p>Tags: protein aggregation; time course; E. coli OP50 to B. subtilis
PXN21; food switching; E. coli OP50; 10.1016/j.celrep.2019.12.078;
NL5901; C. elegans</p>
</div>
</div>
</div>
<p>This is a lot of information!</p>
</section><section id="types-of-metadata"><h2 class="section-heading">Types of metadata<a class="anchor" aria-label="anchor" href="#types-of-metadata"></a>
</h2>
<hr class="half-width">
<p>According to <a href="https://howtofair.dk/" class="external-link">How to FAIR</a>, we can
distinguish between three main types of metadata:</p>
<ul>
<li>
<strong>Administrative metadata:</strong> Data about a project or
resource that is relevant for managing it; for example, project/resource
owner, principal investigator, project collaborators, funder, project
period, etc. They are usually assigned to the data before you collect or
create them.</li>
<li>
<strong>Descriptive or citation metadata:</strong> Data about a
dataset or resource that allows people to discover and identify it; for
example, authors, title, abstract, keywords, persistent identifier,
related publications, etc.</li>
<li>
<strong>Structural metadata:</strong> Data about how a dataset or
resource came about, but also how it is internally structured. For
example, the unit of analysis, collection method, sampling procedure,
sample size, categories, variables, etc. Structural metadata have to be
gathered by the researchers according to best practice in their research
community and will be published together with the data.</li>
</ul>
<p>Descriptive and structural <strong>metadata should be added
continuously</strong> throughout the project.</p>
<div class="section level3">
<h3 id="examples-of-metadata">Examples of metadata<a class="anchor" aria-label="anchor" href="#examples-of-metadata"></a>
</h3>
<p>ere we have an Excel spreadsheet that contains project metadata for a
made-up experiment of plant metabolites:</p>
<p><img src=".././fig/04-metadatafull_spreadsheet_solution.png" alt="Metadata in data table example" class="figure"><br><em>Figure credits: Tomasz Zielinski and Andrés Romanowski</em></p>
<p>Different types of metadata (administrative, descriptive, structural)
are present in this example:</p>
<ul>
<li>Administrative metadata marked in blue</li>
<li>Descriptive metadata marked in orange</li>
<li>Structural metadata marked in green</li>
</ul>
<p>Embedding metadata with your data as in the example is a very useful
technique as the metadata remain joined to the data. As you can see,
typically the metadata can be found in column headers (e.g., Strain,
Media) or as key-value pairs (e.g., Funder, Protocols).</p>
<p>(6 min teaching)</p>
<p>Another form of storing metadata is the creation of a README
file.</p>
<p>For example, check the README files of two datasets:</p>
<ul>
<li><a href="https://github.com/Boehmin/NMJ_analysis" class="external-link">NMJ
analysis</a></li>
<li><a href="https://github.com/BioRDM/COVID-Wastewater-Scotland" class="external-link">COVID
wastewater data</a></li>
</ul>
<p>Check what sections the readme files typically include.</p>
<div id="where-does-data-end-and-metadata-start" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="where-does-data-end-and-metadata-start" class="callout-inner">
<h3 class="callout-title">Where does data end and metadata start?<a class="anchor" aria-label="anchor" href="#where-does-data-end-and-metadata-start"></a>
</h3>
<div class="callout-content">
<p>What is “data” and what is “metadata” can be a matter of perspective:
Some researchers’ metadata can be other researchers’ data.</p>
<p>For example, a funding body is categorized as typical administrative
metadata; however, it can be used to calculate numbers of public
datasets per funder and then used to compare the effects of different
funders’ policies on open practices.</p>
</div>
</div>
</div>
</div>
</section><section id="adding-metadata-to-your-experiments"><h2 class="section-heading">Adding metadata to your experiments<a class="anchor" aria-label="anchor" href="#adding-metadata-to-your-experiments"></a>
</h2>
<hr class="half-width">
<p>Good metadata is crucial for assuring the reusability of your
outcomes. Adding metadata is also a very time-consuming process if done
manually, so collecting metadata should be done incrementally during
your experiment.</p>
<p>As we saw, metadata can take many forms, from as simple as including
a ReadMe.txt file, embedding them inside the Excel files, to using
domain-specific metadata standards and formats.</p>
<p>But, * <strong>What should be included in metadata?</strong> *
<strong>What terms should be used in descriptions?</strong></p>
<div id="exercise-1-what-to-include---discussion" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-1-what-to-include---discussion" class="callout-inner">
<h3 class="callout-title">Exercise 1: What to include - discussion<a class="anchor" aria-label="anchor" href="#exercise-1-what-to-include---discussion"></a>
</h3>
<div class="callout-content">
<p>(5 minutes breakout + 7 discussing results)</p>
<p>Think of the data you generate in your projects, and imagine you are
going to share them.</p>
<p>What information would another researcher need to understand or
reproduce your data (the structural metadata)?</p>
<p>For example, we believe that any dataset should have:</p>
<ul>
<li>a name/title</li>
<li>its purpose or experimental hypothesis</li>
</ul>
<p>Write down and compare your proposals, can we find some common
elements?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1">Show me the solution</h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<p>Some typical elements are:</p>
<ul>
<li>biological material, e.g. Species, Genotypes, Tissue type, Age,
Health conditions</li>
<li>biological context, e.g. speciment growth, entrainment, samples
preparation</li>
<li>experimental factors and conditions, e.g. drug treatments, stress
factors</li>
<li>primers, plasmid sequences, cell line information, plasmid
construction</li>
<li>specifics of data acquisition</li>
<li>specifics of data processing and analysis</li>
<li>definition of variables</li>
<li>accompanying code, software used (version nr), parameters applied,
statistical tests used, seed for randomisation</li>
<li>LOT numbers</li>
</ul>
</div>
</div>
</div>
</div>
</section><section id="minimal-information-standard"><h2 class="section-heading">Minimal Information Standard<a class="anchor" aria-label="anchor" href="#minimal-information-standard"></a>
</h2>
<hr class="half-width">
<p>For many assay methods and experiment types, there are defined
recommendations and guidelines called <strong>Minimal Information
Standards (MIS)</strong>.</p>
<p>The minimum information standard is a set of guidelines for reporting
data derived by relevant methods in the biosciences. Following these
guidelines ensures that the data can be easily:</p>
<ul>
<li><strong>Verified</strong></li>
<li><strong>Analyzed</strong></li>
<li>
<strong>Clearly interpreted</strong> by the wider scientific
community.</li>
</ul>
<p>Adhering to these recommendations also facilitates the foundation
of:</p>
<ul>
<li>Structured databases</li>
<li>Public repositories</li>
<li>Development of data analysis tools</li>
</ul>
<p>Individual minimum information standards are developed by communities
of cross-disciplinary specialists focused on specific methods used in
experimental biology.</p>
<p><strong>Minimum Information for Biological and Biomedical
Investigations (MIBBI)</strong> <a href="https://fairsharing.org/collection/MIBBI" class="external-link">MIBBI</a> is the
collection of the most well-known standards.</p>
<p><img src=".././fig/02-MinimumInformationStandarts.png" alt="Minimum information standards" class="figure"><br>
Figure. Some of the MIBBI minimum information standards from
fairsharing.org</p>
<p><a href="https://fairsharing.org/standards/" class="external-link">FAIRSharing</a> offers
excellent search service for finding standards</p>
<div id="exercise-2-minimal-information-standard-example-5-min" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-2-minimal-information-standard-example-5-min" class="callout-inner">
<h3 class="callout-title">Exercise 2: Minimal information standard
example (5 min)<a class="anchor" aria-label="anchor" href="#exercise-2-minimal-information-standard-example-5-min"></a>
</h3>
<div class="callout-content">
<p>(about 4 + 2 discussion)</p>
<p>Look at the document titled “<strong>Minimum Information About a
Neuroscience Investigation (MINI) Electrophysiology</strong>” (Gibson,
F. et al. Nat Prec (2008): <a href="https://www.nature.com/articles/npre.2008.1720.1.pdf" class="external-link">https://www.nature.com/articles/npre.2008.1720.1.pdf</a>),
which contains recommendations for reporting the use of
electrophysiology in a neuroscience study.<br>
(<em>Neuroscience (or neurobiology) is the scientific study of the
nervous system</em>).</p>
<p>Scroll to <em>Reporting requirement</em> and decide which of the
points 1-8 are:</p>
<ul>
<li><ol style="list-style-type: lower-alpha">
<li>important for understanding and reuse of data</li>
</ol></li>
<li><ol start="2" style="list-style-type: lower-alpha">
<li>important for technical replication</li>
</ol></li>
<li><ol start="3" style="list-style-type: lower-alpha">
<li>could be applied to other experiments in neuroscience</li>
</ol></li>
</ul>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2">Show me the solution</h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" data-bs-parent="#accordionSolution2" aria-labelledby="headingSolution2">
<div class="accordion-body">
<p>Possible answers:</p>
<ul>
<li><ol style="list-style-type: lower-alpha">
<li>2, 3, 4, 5, 6, 8a-b</li>
</ol></li>
<li><ol start="2" style="list-style-type: lower-alpha">
<li>3, 7</li>
</ol></li>
<li><ol start="3" style="list-style-type: lower-alpha">
<li>2, 3, 4, 5, 6</li>
</ol></li>
</ul>
</div>
</div>
</div>
</div>
<p>(3 min)</p>
</section><section id="metadata-and-fair-guidelines"><h2 class="section-heading">Metadata and FAIR guidelines<a class="anchor" aria-label="anchor" href="#metadata-and-fair-guidelines"></a>
</h2>
<hr class="half-width">
<p>Metadata provides extremely valuable information for us and others to
be able to interpret, process, reuse, and reproduce the research data it
accompanies.</p>
<p>Since metadata is data about data, all of the FAIR principles
(Findable, Accessible, Interoperable, and Reusable) apply to it.</p>
<p>Ideally, metadata should be not only machine-readable but also
interoperable so that computer systems can interlink or reason about
it.</p>
</section><section id="attribution"><h2 class="section-heading">Attribution<a class="anchor" aria-label="anchor" href="#attribution"></a>
</h2>
<hr class="half-width">
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">SH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode sh" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a> <span class="ex">Content</span> of this episode was adapted from:</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a> <span class="ex">-</span> [Metadata <span class="at">-</span> FAIR data for climate sciences]<span class="er">(</span><span class="ex">https://escience-academy.github.io/Lesson-FAIR-Data-Climate/metadata/index.html</span><span class="kw">)</span><span class="bu">.</span></span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a> <span class="ex">-</span> [Metadata <span class="at">-</span> How to FAIR]<span class="er">(</span><span class="ex">https://howtofair.dk/how-to-fair/metadata/</span><span class="kw">)</span></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a> <span class="ex">-</span> <span class="pp">[</span><span class="ss">MIBBI</span><span class="pp">]</span><span class="er">(</span><span class="ex">https://fairsharing.org/collection/MIBBI</span><span class="kw">)</span></span></code></pre>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points<a class="anchor" aria-label="anchor" href="#keypoints1"></a>
</h3>
<div class="callout-content">
<ul>
<li>Metadata provides contextual information so that other people can
understand the data.</li>
<li>Metadata is key for data reuse and complying with FAIR
guidelines.</li>
<li>Metadata should be added incrementally through out the project</li>
</ul>
</div>
</div>
</div>
<!-- 
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use. 
 -->
</section></section><section id="aio-03-tidy-data-tables"><p>Content from <a href="03-tidy-data-tables.html">Tidy (meta)data tables</a></p>
<hr>
<p>Last updated on 2024-03-22 |
        
        <a href="https://github.com/BioRDM/fair-for-busy-biologists/edit/main/episodes/03-tidy-data-tables.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 36 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How to represent data in tables</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Do’s and don’ts in Excel</li>
<li>Understand how good table organization aids data reusability.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>(10 min teaching if 1st exc skipped and done as demo)</p>
<p>Tables are a powerful and ubiquitous tool for storing and presenting
information. We encounter them everywhere, from restaurant menus and
bank statements to scientific databases. As biologists, we heavily rely
on tables, often utilizing spreadsheets like Excel.</p>
<p>Excel offers ease of use, flexibility, and power. However, this very
freedom can lead to bad practices that hinder data and metadata
reusability.</p>
<div id="exercise-1-what-can-go-wrong-with-data-in-excel-5-min" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-1-what-can-go-wrong-with-data-in-excel-5-min" class="callout-inner">
<h3 class="callout-title">Exercise 1: What can go wrong with data in
Excel (5 min)<a class="anchor" aria-label="anchor" href="#exercise-1-what-can-go-wrong-with-data-in-excel-5-min"></a>
</h3>
<div class="callout-content">
<p>We suggest the instuctor to go through problems, instead</p>
<p>Have a look at the example <a href=".././files/04-bad-metadata.xlsx">data-file</a> in Excel.</p>
<figure><img src=".././fig/bad-metadata.png" alt="bad-metadata" class="figure mx-auto d-block"><div class="figcaption">bad-metadata</div>
</figure><p>This data resembles plant experiment results under various light
conditions. Imagine you’re tasked with working on this file.</p>
<ol style="list-style-type: decimal">
<li>What aspects of this table appear confusing?</li>
<li>What clarifications would you seek from the original author before
using this data?</li>
<li>How would calculating average biomass or biomass per genotype be
problematic with this format?</li>
<li>Advanced data analysis often involves programmatic manipulation.
This requires converting the data to a text format like CSV or TSV, or
using libraries that can “read and convert Excel files on the fly.” Save
this file in a text format, close Excel, and then reopen the saved file.
What changes occur?</li>
</ol>
<p>Have you encountered similar tables in your experience? Do you
believe this example realistically reflects common issues?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1">Show me the solution</h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<p>This file hopefully unrealistically exacerbates typical bad practices
in Excel. 1+2. Some things that may be confusing:</p>
<ul>
<li>Why are there two tables, are the period measurement related to the
metabolics i.e. same samples?</li>
<li>Do colors in the period table have the same meaning? Seems no.</li>
<li>Why can row 22 be read, whilst row 13 says error?</li>
<li>What is the meaning of values in the media column?</li>
<li>Are the genotypes the same in different blocks or not?</li>
<li>What is the meaning behind bold text in the table?</li>
<li>What is the definition of the terms/why are units
missing/inconsistent?</li>
</ul>
<ol start="3" style="list-style-type: decimal">
<li><em>calculations</em></li>
</ol>
<ul>
<li>Biomass weights must be in the same unit before averaging. Any text
entries need to be replaced with the appropriate value.</li>
<li>Averaging biomass per genotype requires manually selecting relevant
entries.</li>
</ul>
<ol start="4" style="list-style-type: decimal">
<li><em>saving to txt</em></li>
</ol>
<ul>
<li>Information about light conditions is entirely lost.</li>
<li>Header column order might be scrambled.</li>
<li>The update date format may change meaning depending on location
(e.g., switching year and day order).</li>
</ul>
</div>
</div>
</div>
</div>
<section id="common-spreadsheet-errors"><h2 class="section-heading">Common Spreadsheet Errors<a class="anchor" aria-label="anchor" href="#common-spreadsheet-errors"></a>
</h2>
<hr class="half-width">
<p>(25 min teaching)</p>
<div class="section level3">
<h3 id="using-multiple-tables">
<a name="tables"></a> 1. Using multiple tables<a class="anchor" aria-label="anchor" href="#using-multiple-tables"></a>
</h3>
<p>Multiple data tables within one spreadsheet can confuse both humans
and computers:</p>
<ul>
<li>For humans, they can prompt finding associations where they are not
present.</li>
<li>For computers, each spreadsheet row is typically treated as one
observation.</li>
</ul>
<p>Additionally, you may potentially use the same field name in multiple
places.</p>
<p>In our example, both column A and K named <code>sample</code>
represent different information, and values in row 6 from both parts are
not related.</p>
</div>
<div class="section level3">
<h3 id="using-multiple-tabs">
<a name="tabs"></a> 2. Using multiple tabs<a class="anchor" aria-label="anchor" href="#using-multiple-tabs"></a>
</h3>
<p>Using multiple tabs might seem like an easy way to organize data, but
it has its drawbacks:</p>
<ul>
<li>Tabs make it more difficult to perform programmatic analysis.</li>
<li>Tabs can be ignored (under the radar) when opening files. For
example, the recently used tab is shown first, and other users may not
notice that there are many other tabs before it (we have seen this
happen frequently when importing data).</li>
</ul>
<p>However,</p>
<ul>
<li>Tabs seem like a perfect place to add administrative, descriptive
metadata so they are next to data without “polluting” the data
table.</li>
<li>Having secondary data next to primary in subsequent tabs permits
easy inspection of all without a need to always send a collection of
files.</li>
</ul>
<p>Our recommendation: Use tabs with caution; the more automatic
analysis you do, the less frequently you should use tabs.</p>
</div>
<div class="section level3">
<h3 id="not-filling-in-zeros">
<a name="zeros"></a> 3. Not filling in zeros<a class="anchor" aria-label="anchor" href="#not-filling-in-zeros"></a>
</h3>
<p>You might think that when you’re measuring something and it’s usually
a zero, like the number of times a rabbit is observed in a survey, why
bother writing in the number zero in that column when it’s mostly
zeros?</p>
<p>However, there’s a difference between a zero and a blank cell in a
spreadsheet. To the computer, a zero is data - you measured or counted
it. A blank cell means that it wasn’t measured, and the computer will
interpret it as an unknown value (otherwise known as a null value).</p>
<p>Spreadsheets or statistical programs will likely misinterpret blank
cells that you intend to be zeros.</p>
<p>Because of this, it’s very important to record zeros as zeros and
truly missing data as nulls.</p>
</div>
<div class="section level3">
<h3 id="using-problematic-null-values">
<a name="null"></a> 4. Using problematic null values<a class="anchor" aria-label="anchor" href="#using-problematic-null-values"></a>
</h3>
<p>For example: using -999, -1, or 0 (other numerical values) to
represent missing data.</p>
<p>Other times, different null values are used to convey different
reasons why the data isn’t there. For example, -1 for not recorded, -2
for contamination, etc.</p>
<p>This is important information to capture, but effectively uses one
column to capture two pieces (real values and comment or status). It
would be good here to create a new column like ‘data_missing’ and use
that column to capture the different reasons.</p>
<p>Whatever the reason, it’s a problem if unknown or missing data is
recorded as -999, 999, or 0. Many statistical programs will not
recognize that these are intended to represent missing (null)
values.</p>
<p>It is essential to use a clearly defined and consistent null
indicator. Blanks (most applications) and NA (for R) are good
choices.</p>
<p>White et al. (2013) explain good choices for indicating null values
for different software applications in their article: <a href="https://peerj.com/preprints/7/" class="external-link">Nine simple ways to make it easier
to (re)use your data.</a> Ideas in Ecology and Evolution.</p>
<figure><img src=".././fig/null_values_table_1.jpg" alt="White et al." class="figure mx-auto d-block"><div class="figcaption">White et al.</div>
</figure>
</div>
<div class="section level3">
<h3 id="using-formatting-to-convey-information-organizing-data">
<a name="formatting"></a> 5. Using formatting to convey information
organizing data<a class="anchor" aria-label="anchor" href="#using-formatting-to-convey-information-organizing-data"></a>
</h3>
<p>Never highlight cells, rows, or columns that should be excluded from
an analysis, or to mark particular properties/conditions.</p>
<p>In our example file, information about light conditions is only
encoded as a color. Formatting information is not available to analysis
software and almost certainly will be lost during processing.</p>
<p>All the information should be encoded as a field value, for example,
in columns like:<br>
condition, calibrated,<br>
validated, omitted, etc.</p>
<p>You can still use colors/fonts to help with readability (just make
sure they help and don’t distract) but no information should be lost if
data is exported to plain text.</p>
<p>If you are not careful, formatting a worksheet to be more
aesthetically pleasing can compromise your computer’s ability to see
associations in the data.</p>
<p><strong>Never merge cells, as it will make your data unreadable by
statistics software</strong>.</p>
<p>Don’t leave blank rows to indicate separations in data</p>
</div>
<div class="section level3">
<h3 id="placing-comments-or-units-in-cells">
<a name="units"></a> 6. Placing comments or units in cells<a class="anchor" aria-label="anchor" href="#placing-comments-or-units-in-cells"></a>
</h3>
<p>Most analysis software can’t see Excel or LibreOffice comments, and
would be confused by comments placed within your data cells.</p>
<p>As described above for formatting, create another field if you need
to add notes to cells.</p>
<p>Similarly, don’t include units in cells: ideally, all the
measurements you place in one column should have the same unit, but if
for some reason they don’t, create another field and specify the unit. A
field value <code>0.123 g</code> will need to be parsed by a script into
<code>0.123</code> and <code>g</code> in order to be used for
calculations, don’t add extra work for others.</p>
</div>
<div class="section level3">
<h3 id="entering-more-than-one-piece-of-information-in-a-cell">
<a name="info"></a> 7. Entering more than one piece of information
in a cell<a class="anchor" aria-label="anchor" href="#entering-more-than-one-piece-of-information-in-a-cell"></a>
</h3>
<p>Don’t include more than one piece of information in a cell (like
<code>M30</code>, <code>F25</code>, for male age 30 and female 25, or
<code>Red light + Ampicilin + High nitrogen</code>).</p>
<p>This will limit the ways in which you can analyze your data. Design
your data sheet to include this information. For example, include one
column for age of individuals and a separate column for sex.</p>
</div>
<div class="section level3">
<h3 id="inconsistency-in-used-values">
<a name="consistency"></a> 8. Inconsistency in used values<a class="anchor" aria-label="anchor" href="#inconsistency-in-used-values"></a>
</h3>
<p>Using “synonyms” or alternatives to describe the same values.</p>
<p>For example: <code>E. Coli</code>, <code>EColi</code>,
<code>Escherichia coli</code> in the same table</p>
</div>
<div class="section level3">
<h3 id="using-problematic-field-names">
<a name="field_name"></a> 9. Using problematic field names<a class="anchor" aria-label="anchor" href="#using-problematic-field-names"></a>
</h3>
<p>Choose descriptive field names, but be careful not to include spaces,
numbers, or <strong>special characters</strong> of any kind (including
national characters: óęłńöüä). Spaces can be misinterpreted by parsers
that use whitespaces as delimiters and some programs don’t like field
names that are text strings starting with numbers (e.g. Excel).</p>
<p>Underscores (<code>_</code>) are a good alternative to spaces.
Consider writing names in camel case (like this: ExampleFileName) to
improve readability.</p>
<p>Remember that abbreviations that make sense at the moment may not be
so obvious in 6 months, but don’t overdo it with names that are
excessively long.</p>
<p>Including the units in the field names avoids confusion and enables
others to readily interpret your fields. Or add a separate row to help
automatic unit conversions.</p>
<p><strong>Examples</strong></p>
<table align="left" style="width =50%; border: 1px solid black;" class="table">
<tr>
<td>
<b>Good Name</b>
</td>
<br><td>
<b>Good Alternative </b>
</td>
<br><td>
<b>Avoid </b>
</td>
<br>
</tr>
<tr>
<td>
Max_temp_C
</td>
<td>
MaxTemp
</td>
<td>
Maximum Temp (°C)
</td>
</tr>
<tr>
<td>
Precipitation_mm
</td>
<td>
Precipitation
</td>
<td>
precmm
</td>
</tr>
<tr>
<td>
Mean_year_growth
</td>
<td>
MeanYearGrowth
</td>
<td>
Mean growth/year
</td>
</tr>
<tr>
<td>
sex
</td>
<td>
sex
</td>
<td>
M/F
</td>
</tr>
<tr>
<td>
weight
</td>
<td>
weight
</td>
<td>
w.
</td>
</tr>
<tr>
<td>
cell_type
</td>
<td>
CellType
</td>
<td>
Cell Type
</td>
</tr>
<tr>
<td>
Observation_01
</td>
<td>
first_observation
</td>
<td>
1st Obs
</td>
</tr>
</table>
</div>
<div class="section level3">
<h3 id="using-special-characters-in-data">
<a name="special"></a> 10. Using special characters in data<a class="anchor" aria-label="anchor" href="#using-special-characters-in-data"></a>
</h3>
<p>For example, when writing longer text in a cell, people often include
line breaks, em-dashes, etc in their spreadsheet. Also, when copying
data from applications such as Word, formatting and ‘fancy’ non-standard
characters (such as left- and right-aligned quotation marks) are
included. When exporting this data into a coding/statistical environment
or into a relational database, dangerous things may occur, such as lines
being cut in half and encoding errors being thrown.</p>
<p>General best practice is to avoid adding characters such as new
lines, tabs, vertical tabs and sadly, national characters. In other
words, treat a text cell as if it were a simple web form that can only
contain text and spaces.</p>
</div>
<div class="section level3">
<h3 id="values-without-field-labels">
<a name="metadata"></a> 11. Values without field labels<a class="anchor" aria-label="anchor" href="#values-without-field-labels"></a>
</h3>
<p>It may be obvious for us that a cell
<code>Ann Smart and Daniele Hardwork</code> contains authors and
<code>Starch content in Arabidopis T. under different light conditions</code>
is a title, but this is not obvious for a computer program.</p>
<p>Always label values as the column header, or as a first field on the
left:</p>
<pre><code>Author  |  Ann Smart  |  Daniele Hardwork
Title   |  Starch content in Arabidopis T. under different light conditions</code></pre>
<div id="exercise-2-spotting-problems" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-2-spotting-problems" class="callout-inner">
<h3 class="callout-title">Exercise 2: Spotting problems<a class="anchor" aria-label="anchor" href="#exercise-2-spotting-problems"></a>
</h3>
<div class="callout-content">
<p>(5 min + 4 talking)</p>
<p>Look at the following rows and columns in the problematic table:</p>
<ul>
<li>Row 5</li>
<li>Row 2</li>
<li>Column C</li>
<li>Column E</li>
<li>Column L <img src=".././fig/bad-metadata.png" alt="bad-metadata" class="figure">
</li>
</ul>
<p>which of the problems discussed above can you spot in these rows and
column?</p>
<p>Here, we list them again:</p>
<ol style="list-style-type: decimal">
<li>Using multiple tables</li>
<li>Using multiple tabs</li>
<li>Not filling in zeros</li>
<li>Using problematic null values</li>
<li>Using formatting to convey information and organizing data</li>
<li>Placing comments or units in cells</li>
<li>Entering more than one piece of information in a cell</li>
<li>Inconsistency in used values</li>
<li>Using problematic field names</li>
<li>Using special characters in data</li>
<li>Values without field labels</li>
</ol>
<p>Type the problem number(s) next to the table elements</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2">Show me the solution</h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<ul>
<li>Row 5: 5, 9</li>
<li>Row 2: 11, 7</li>
<li>Column C: 8</li>
<li>Column E: 6, 5</li>
<li>Column L: 3</li>
</ul>
</div>
</div>
</div>
</div>
</div>
</section><section id="clean-data-tables-make-life-easier"><h2 class="section-heading">Clean data tables make life easier<a class="anchor" aria-label="anchor" href="#clean-data-tables-make-life-easier"></a>
</h2>
<hr class="half-width">
<p>(2 min teaching)</p>
<p>Let’s check the cleaned version of the previous <a href=".././files/04-better-metadata.xlsx">file</a>.</p>
<p>Look how easy it is to calculate the average biomas, and if you have
Office365, see how easy it is to get average values per genotype.</p>
<p>Do you think it would take more effort to record data in a cleaned
way to start with, in comparison to the ‘bad example’ presented
before?</p>
<p>How long do you think it took to “clean” the original, problematic
data?</p>
<div id="exercise-3-outsmarted-by-excel-4-min" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="exercise-3-outsmarted-by-excel-4-min" class="callout-inner">
<h3 class="callout-title">Exercise 3: Outsmarted by Excel (4 min)<a class="anchor" aria-label="anchor" href="#exercise-3-outsmarted-by-excel-4-min"></a>
</h3>
<div class="callout-content">
<p>Excel has built in autoformatting functions which can make
biologist’s life more difficult</p>
<p>## Finding the right date</p>
<p>Open Excel and type following values into the cells:</p>
<table class="table">
<thead><tr class="header">
<th>A</th>
<th>B</th>
<th>C</th>
<th>D</th>
<th>E</th>
<th>F</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Gene</td>
<td>SEPT2</td>
<td>Sample</td>
<td>0013</td>
<td>Record</td>
<td>12/5/4</td>
</tr>
<tr class="even">
<td>Mar1</td>
<td>1 March</td>
<td>Mar-1</td>
<td>1-3</td>
<td>14/3/20</td>
<td>43904</td>
</tr>
</tbody>
</table>
<ul>
<li>Is what you see what you typed?</li>
<li>Can you force Excel to keep your formatting?</li>
<li>Do you know which year the dates represent?</li>
</ul>
</div>
</div>
</div>
<p>(7 min teaching)</p>
<p>A team of Australian researchers analyzed nearly 3,600 genetics
papers <a href="https://doi.org/10.1186/s13059-016-1044-7" class="external-link">Ziemann
2016</a>. As is common practice in the field, these papers all came with
supplementary files containing lists of genes used in research.</p>
<p>The Australian researchers found that roughly 1 in 5 of these papers
included errors in their gene lists that were due to Excel automatically
converting gene names to things like calendar dates or random
numbers.</p>
<p>Those errors actually forced the renaming of 27 gene symbols
including SEPT4 (now SEPTIN4) and MARCH1 (now MARCHF1).</p>
</section><section id="handling-dates"><h2 class="section-heading">Handling dates<a class="anchor" aria-label="anchor" href="#handling-dates"></a>
</h2>
<hr class="half-width">
<p>Storing and handling dates can be problematic, even in programming
languages, as we often encounter dates in various formats that are
region-specific.</p>
<p>Have you ever been confused by a meeting date from an American
collaborator?</p>
<p>When dealing with dates that represent “real” data, such as sample
collection from a patient or field measurements, preventing
misinterpretations is crucial. There are two safe options:</p>
<div class="section level4">
<h4 id="store-dates-in-three-columns-for-year-month-and-day">Store dates in three columns for year, month, and day:<a class="anchor" aria-label="anchor" href="#store-dates-in-three-columns-for-year-month-and-day"></a>
</h4>
<table class="table">
<thead><tr class="header">
<th>Year</th>
<th>Month</th>
<th>Day</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>2021</td>
<td>3</td>
<td>16</td>
</tr>
<tr class="even">
<td>2021</td>
<td>10</td>
<td>1</td>
</tr>
</tbody>
</table>
</div>
<div class="section level4">
<h4 id="or-store-the-date-as-an-iso-string">Or store the date as an <a href="https://en.wikipedia.org/wiki/ISO_8601" class="external-link">ISO</a> string:<a class="anchor" aria-label="anchor" href="#or-store-the-date-as-an-iso-string"></a>
</h4>
<p><code>YYYYMMDD</code> e.g., <code>20210316</code><br><code>YYYY-MM-DD</code> e.g., <code>2021-03-16</code></p>
<p>Even though the <code>YYYY-MM-DD</code> format is normally preferred,
it may unfortunately be reformatted by Excel according to your locale
when saving to text like .csv!</p>
<p>When using text files (.csv, .tsv), you should always document the
format you are using to represent dates.</p>
<div id="to-use-or-not-to-use-excel" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="to-use-or-not-to-use-excel" class="callout-inner">
<h3 class="callout-title">To Use or Not to Use Excel<a class="anchor" aria-label="anchor" href="#to-use-or-not-to-use-excel"></a>
</h3>
<div class="callout-content">
<p>The Excel file format <code>.xlsx</code> is now open, widely used,
and supported by external libraries, making it considered interoperable.
Nowadays, it is admissible as being FAIR.</p>
<p>However, plain text files like comma or tab-separated values (.csv,
.tsv) can be accessed without any special software. Data in a CSV file
can also be easily imported into other formats and environments, such as
SQLite and R. We are not tied to a certain version of a certain
expensive program when we work with CSV files, so it is a good format to
work with for maximum portability, interoperability, and endurance.</p>
<p>If such files are handled only with text editors or programmatically
(R, Python), then they are a safer option as they prevent the
autoformatting issues described before.</p>
<p>If you analyze your data with R or Python, or you know that your data
are meant to be processed that way, you should be using text formats
whenever possible, and as soon as you capture your data.</p>
<p>However, if you are used to only using Excel and so does your
community, just keep using it. Just be aware of the possible pitfalls
discussed, especially when working with gene or protein names and
accession numbers.</p>
</div>
</div>
</div>
<div id="cleaning-data-with-open-refine" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="cleaning-data-with-open-refine" class="callout-inner">
<h3 class="callout-title">Cleaning data with Open Refine<a class="anchor" aria-label="anchor" href="#cleaning-data-with-open-refine"></a>
</h3>
<div class="callout-content">
<p>Data cleaning and reorganization can be time-consuming, but there are
tools to make it easier.</p>
<p><a href="https://openrefine.org/" class="external-link">OpenRefine</a> is a powerful option
for tackling messy data. It can:</p>
<ul>
<li>Clean your data: Remove errors and inconsistencies.</li>
<li>Transform your data: Convert it between different formats.</li>
<li>Extend your data: Enrich it with web services and external
sources.</li>
</ul>
<p>OpenRefine lets you tackle common data quality issues. For example,
you can:</p>
<ul>
<li>Merge synonyms: Standardize terms like <code>"E. Coli"</code>,
<code>"EColi"</code>, and <code>"Escherichia coli"</code> into a single
term.</li>
<li>Split values: Separate entries in a <code>"Name"</code> field into
separate <code>"FirstName"</code> and <code>"LastName"</code>
fields.</li>
</ul>
<p><strong>Want to learn more?</strong></p>
<p>Check out this carpentry course specifically designed for ecologists:
<strong>Data Cleaning with OpenRefine for Ecologists: <a href="https://datacarpentry.org/OpenRefine-ecology-lesson/" class="external-link">https://datacarpentry.org/OpenRefine-ecology-lesson/</a></strong></p>
</div>
</div>
</div>
<div id="exercise-4-data-tables-and-fair-4-min" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-4-data-tables-and-fair-4-min" class="callout-inner">
<h3 class="callout-title">Exercise 4: Data tables and FAIR (4 min)<a class="anchor" aria-label="anchor" href="#exercise-4-data-tables-and-fair-4-min"></a>
</h3>
<div class="callout-content">
<p>Which of the following statements is true/false (T or F):</p>
<ul>
<li>Do’s and don’ts help in programmatic re-use:</li>
<li>Avoiding multiple tabs improves interoperability:</li>
<li>Having accompanying README file with a table description is not
FAIR:</li>
<li>No ‘spaces’ in columns headers improve readability:</li>
<li>2022-07-15 is ISO date format:</li>
<li>20220715 date format is better for excel than 2022-07-15:</li>
<li>“No data” is better than leaving cell “blank” for missing data:</li>
</ul>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3">Show me the solution</h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" aria-labelledby="headingSolution3" data-bs-parent="#accordionSolution3">
<div class="accordion-body">
<ul>
<li>Do’s and don’ts help in programmatic re-use: T</li>
<li>Avoiding multiple tabs improves interoperability: T</li>
<li>Having accompanying README file with a table description is not
FAIR: F</li>
<li>No ‘spaces’ in columns headers improve readability: F</li>
<li>2022-07-15 is ISO date format: T</li>
<li>20220715 date format is better for excel than 2022-07-15: T</li>
<li>“No data” is better than leaving cell “blank” for missing data:
F</li>
</ul>
</div>
</div>
</div>
</div>
<div class="keypoints:">
<ul>
<li>Never use formatting to encode information</li>
<li>Include only one piece of information in a cell</li>
<li>It is easier to store data in the correct form than to clean data
for reuse</li>
</ul>
</div>
<!-- 
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use. 
 -->
</div>
</section></section><section id="aio-04-files-organization"><p>Content from <a href="04-files-organization.html">Working with files</a></p>
<hr>
<p>Last updated on 2024-03-22 |
        
        <a href="https://github.com/BioRDM/fair-for-busy-biologists/edit/main/episodes/04-files-organization.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 50 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How should I name my files?</li>
<li>How does folder organisation help me</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Understand elements of good naming strategy</li>
<li>Evaluate pros and cons of different project organizations</li>
<li>Explain how file management helps in being FAIR</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section id="project-organization-planning-file-names-and-folder-structure"><h2 class="section-heading">Project organization: planning file names and folder structure<a class="anchor" aria-label="anchor" href="#project-organization-planning-file-names-and-folder-structure"></a>
</h2>
<hr class="half-width">
<p>(3 min teaching)</p>
<p>Before you even start collecting or working with data you should
decide how to structure and name files and folders. This will:</p>
<ul>
<li>allow for standardized data collection and analysis by many team
members.</li>
<li>make it easier for the researcher to determine where files should be
saved.</li>
<li>avoid file duplication.</li>
<li>help make retrieval and archiving more efficient.</li>
</ul>
<p><img src=".././fig/07-intro_folder_structure.png" alt="Intro to folder structure" class="figure"><em>Figure credits: Andrés
Romanowski</em></p>
<p>Consistent naming and organization of files in folders has two main
goals:</p>
<ul>
<li>quick to find files</li>
<li>ability to tell the file content without opening it</li>
</ul></section><section id="naming-your-files-and-folders"><h2 class="section-heading">Naming your files (and folders)<a class="anchor" aria-label="anchor" href="#naming-your-files-and-folders"></a>
</h2>
<hr class="half-width">
<p>One important and often overlooked practice of organizing, sharing,
and keeping track of data files is standardized naming.<br>
It is important to develop naming conventions which permit the encoding
of experimental factors which are important to the project.</p>
<p>File (folder) names should be consistent, meaningful to you and your
collaborators, allow you to easily find what you are looking for, give
you a sense of the content without opening the file, and allow you to
easily identify if something is missing.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">SH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode sh" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="ex">Example</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="kw">`</span><span class="ex">LD_phyA_off_t04_2020-08-12.norm.xlsx</span><span class="kw">`</span></span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="ex">could</span> be a file that contains:</span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a><span class="ex">*</span> normalized data <span class="er">(</span><span class="ex">norm</span><span class="kw">)</span><span class="ex">,</span> </span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a><span class="ex">*</span> from experiment in long day <span class="er">(</span><span class="ex">LD</span><span class="kw">)</span> </span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a><span class="ex">*</span> for genotype phyA</span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a><span class="ex">*</span> with media without sucrose <span class="er">(</span><span class="ex">off</span><span class="kw">)</span></span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a><span class="ex">*</span> at timepoint 4 <span class="er">(</span><span class="ex">t04</span><span class="kw">)</span></span></code></pre>
</div>
<div id="exercise-1-naming-and-sorting-12-min---3-min-explanation" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-1-naming-and-sorting-12-min---3-min-explanation" class="callout-inner">
<h3 class="callout-title">Exercise 1: Naming and sorting (12 min - 3 min
explanation)<a class="anchor" aria-label="anchor" href="#exercise-1-naming-and-sorting-12-min---3-min-explanation"></a>
</h3>
<div class="callout-content">
<p>(5 min breakout + 5 discussing)</p>
<p>Have a look at the example files from a project, similar to the one
from the previous metadata episode.</p>
<p>All the files have been sorted by name and demonstrate consequences
of different naming strategies.</p>
<p>For your information, to encode experimental details the following
conventions were taken:</p>
<ul>
<li>phyB/phyA are sample genotypes</li>
<li>sXX is the sample number</li>
<li>LD/SD are different light conditions (long or short day)</li>
<li>on/off are different media (on sucrose, off sucrose)</li>
<li>measurement date</li>
<li>other details are timepoint and raw or normalized data</li>
</ul>
<pre><code>2020-07-14_s12_phyB_on_SD_t04.raw.xlsx  
2020-07-14_s1_phyA_on_LD_t05.raw.xlsx  
2020-07-14_s2_phyB_on_SD_t11.raw.xlsx  
2020-08-12_s03_phyA_on_LD_t03.raw.xlsx  
2020-08-12_s12_phyB_on_LD_t01.raw.xlsx  
2020-08-13_s01_phyB_on_SD_t02.raw.xlsx  
2020-7-12_s2_phyB_on_SD_t01.raw.xlsx  
AUG-13_phyB_on_LD_s1_t11.raw.xlsx  
JUL-31_phyB_on_LD_s1_t03.raw.xlsx  
LD_phyA_off_t04_2020-08-12.norm.xlsx  
LD_phyA_on_t04_2020-07-14.norm.xlsx  
LD_phyB_off_t04_2020-08-12.norm.xlsx  
LD_phyB_on_t04_2020-07-14.norm.xlsx  
SD_phyB_off_t04_2020-08-13.norm.xlsx  
SD_phyB_on_t04_2020-07-12.norm.xlsx  
SD_phya_off_t04_2020-08-13.norm.xlsx  
SD_phya_ons_t04_2020-07-12.norm.xlsx  
ld_phyA_ons_t04_2020-08-12.norm.xlsx  </code></pre>
<ul>
<li>What are the problems with having the date first?</li>
<li>How do different date formats behave once sorted?</li>
<li>Can you tell the importance of a leading 0 (zeros)?</li>
<li>Is it equally easy to find all data from LD conditions as ON
media?</li>
<li>Can you spot the problem when using different cases
(upper/lower)?</li>
<li>Do you see benefits of keeping consistent lengths of the naming
conventions?</li>
<li>Do you see what happens when you mix conventions?</li>
</ul>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1">Show me the solution</h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<ul>
<li>Using dates up front makes it difficult to quickly find data for
particular conditions or genotypes. It also masks the “logical” order of
samples or timepoints.</li>
<li>Named months break the “expected” sorting, same as dates without
leading 0</li>
<li>Without leading zeros, ‘s12’ appear before s1 and s2</li>
<li>the first (and second) part of the name are easiest to spot</li>
<li>the last file is also from LD conditions, but appears after SD, same
with ‘phya’ genotypes</li>
<li>the last 3 file names are easiest to read as all parts appear on top
of each other due to the same 3 letter-length codes ons and off</li>
<li>The lack of consistency makes it very difficult to get data from
related samples/conditions.</li>
</ul>
</div>
</div>
</div>
</div>
<p>(7 min teaching)</p>
<p>Some things to take into account to decide on your naming convention
are:</p>
<ul>
<li>Does your convention make your files easy to sort and find (e.g. by
important features)?</li>
<li>Include parameters that are as descriptive as possible (i.e.:
project, experiment, researcher, sample, organism, date/range, data
type, method).</li>
<li>Defined a standard vocabulary (shortcuts) for parameters and
document any abbreviation.</li>
<li>Decide which elements go in which order.</li>
<li>Decide the convention when to use symbols, capitals, hyphens (e.g
kebab-case, CamelCase, or snake_case).</li>
<li>Define a maximum name length. Aim for filenames no longer than ~30
characters.</li>
</ul>
<p><strong>Do’s:</strong></p>
<ul>
<li>for dates use the YYYY-MM-DD standard and place at the end of the
file UNLESS you need to organize your files chronologically</li>
<li>include version number (if applicable), use leading zeroes (i.e.:
v005 instead of v5).</li>
<li>make sure the 3-letter file format extension is present at the end
of the name (e.g. .doc, .xls, .mov, .tif)</li>
<li>add a <strong>PROJECT_STRUCTURE</strong> (README) file in your top
directory which details your naming convention, directory structure and
abbreviations</li>
</ul>
<p><strong>Don’ts:</strong></p>
<ul>
<li>avoid using spaces (use _ or - instead)</li>
<li>avoid dots, commas and special characters (e.g. ~ ! @ # $ % ^ &amp;
* ( ) ` ; &lt; &gt; ? , [ ] { } ‘ “)</li>
<li>avoid using language specific characters (e.g óężé), unfortunately
they still cause problems with most software or between operating
systems (OS)</li>
<li>avoid long names</li>
<li>avoid repetition, e.g if directory name is
<em>Electron_Microscopy_Images</em>, and file
<em>ELN_MI_IMG_20200101.img</em> then ELN_MI_IMG is redundant</li>
<li>avoid deep paths with long names (i.e. deeply nested folders with
long names) as archiving or moving between OS may fail</li>
</ul>
<div id="challenge2" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Challenge<a class="anchor" aria-label="anchor" href="#challenge2"></a>
</h3>
<div class="callout-content">
<p>## Exercise 2: A good name (3 min)</p>
<p>Select which file options adhere the best to the presented
recommendations:</p>
<ol style="list-style-type: decimal"><li>
</li></ol>
<ol style="list-style-type: lower-alpha">
<li>analysis-20210906.xlsx<br>
</li>
<li>rna-levels-by-site.v002.xlsx<br>
</li>
<li>analysis of rna levels from 5Aug2021.xlsx</li>
</ol>
<ol start="2" style="list-style-type: decimal"><li>
</li></ol>
<ol style="list-style-type: lower-alpha">
<li>20210906-birds-count-EDI.csv<br>
</li>
<li>birds.csv<br>
</li>
<li>birds-count&amp;diversity EDI 2021-09-06.csv</li>
</ol>
<ol start="3" style="list-style-type: decimal"><li>
</li></ol>
<ol style="list-style-type: lower-alpha">
<li>2020-7-12_s2_phyB_+_SD_t01.raw.xlsx<br>
</li>
<li>ld_phyA_on_s02-t01_2020-07-12.norm.xlsx<br>
</li>
<li>ld_phya_ons_02-01_2020-07-12.norm.xlsx</li>
</ol>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2">Show me the solution</h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<ul>
<li>1 b)</li>
<li>2 a)</li>
<li>3 b)</li>
</ul>
</div>
</div>
</div>
</div>
<p>If adding all the relevant details to file names makes them too long,
it is often a signal that you should use folders to organize the files
and capture some of those parameters.</p>
<div id="exercise-3-folders-vs-files" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-3-folders-vs-files" class="callout-inner">
<h3 class="callout-title">Exercise 3: Folders vs Files<a class="anchor" aria-label="anchor" href="#exercise-3-folders-vs-files"></a>
</h3>
<div class="callout-content">
<p>(5 min)</p>
<p>Have a look as these two different organization strategies:</p>
<ol style="list-style-type: decimal">
<li><p>|– Project<br>
|– |– arab_LD_phyA_off_t04_2020-08-12.metab.xlsx</p></li>
<li><p>|– Project<br>
|– |– arabidopsis<br>
|– |– |– long_day<br>
|– |– |– |– phyA<br>
|– |– |– |– |– off_sucrose_2020-08-12<br>
|– |– |– |– |– |– t04.metab.xlsx</p></li>
</ol>
<p>Can you think of scenarios in which one is better suited than other?
Hint: think of other files that could be present as well.</p>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3">Show me the solution</h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" aria-labelledby="headingSolution3" data-bs-parent="#accordionSolution3">
<div class="accordion-body">
<p>The first strategies, can work very well if the project has only few
files, so all of them can quickly be accessed (no need to change
folders) and the different parameters are easily visible. For example a
couple of conditions, couple of genotypes or species</p>
<p>|– Project<br>
|– |– arab_LD_phyA_off_t04_2020-08-12.metab.xlsx<br>
|– |– arab_LD_WILD_off_t03_2020-08-11.metab.xlsx<br>
|– |– arab_SD_phyA_off_t01_2020-05-12.metab.xlsx<br>
|– |– arab_SD_WILD_off_t02_2020-05-11.metab.xlsx<br>
|– |– rice_LD_phyA_off_t05_2020-05-02.metab.xlsx<br>
|– |– rice_LD_WILD_off_t06_2020-05-02.metab.xlsx<br>
|– |– rice_SD_phyA_off_t07_2020-06-02.metab.xlsx<br>
|– |– rice_SD_WILD_off_t08_2020-06-02.metab.xlsx</p>
<p>The second strategy works better if we have a lot of individual files
for each parameter. For example, imagine the metabolites are measured
hourly throughout the day, and there are ten different genotypes, two
species and 4 light conditions. You would not want to have all the 2000
files in one folder.</p>
<p>|– Project<br>
|– |– arabidopsis<br>
|– |– |– long_day<br>
|– |– |– |– phyA<br>
|– |– |– |– |– off_sucrose_2020-08-12<br>
|– |– |– |– |– |– t01.metab.xlsx<br>
|– |– |– |– |– |– t02.metab.xlsx<br>
|– |– |– |– |– |– t03.metab.xlsx<br>
|– |– |– |– |– |– …<br>
|– |– |– |– |– |– t23.metab.xlsx<br>
|– |– |– |– |– |– t24.metab.xlsx<br>
|– |– rice<br>
|– |– |– long_day<br>
|– |– |– |– phyA<br>
|– |– |– |– |– off_sucrose_2020-06-03<br>
|– |– |– |– |– |– t01.metab.xlsx<br>
|– |– |– |– |– |– …<br>
|– |– |– |– |– |– t24.metab.xlsx</p>
</div>
</div>
</div>
</div>
<p>(4 min teaching)</p>
</section><section id="must-do-document-your-strategy"><h2 class="section-heading">Must do: Document your strategy<a class="anchor" aria-label="anchor" href="#must-do-document-your-strategy"></a>
</h2>
<hr class="half-width">
<p>Regardless of whether you are using long filenames or incorporating
some of the variables within the folder structure, document it!<br>
Always include a PROJECT_STRUCTURE (or README) file describing your file
naming and folder organisation conventions.</p>
</section><section id="strategies-to-set-up-a-clear-folder-structure"><h2 class="section-heading">Strategies to set up a clear folder structure<a class="anchor" aria-label="anchor" href="#strategies-to-set-up-a-clear-folder-structure"></a>
</h2>
<hr class="half-width">
<p>(3 min teaching)</p>
<p>Establishing a system that allows you to access your files, avoid
duplication and ensure that your data can be easily found needs
planning.</p>
<p>You can start by developing a logical folder structure. To do so, you
need to take into account the following suggestions:</p>
<ul>
<li>Use folders to group related files. A single folder will make it
easy to locate them.</li>
<li>Name folders appropriately: use descriptive names after the areas of
work to which they relate.</li>
<li>Structure folders hierarchically: use broader topics for your main
folders and increase in specificity as you go down the hierarchy.</li>
<li>Be consistent: agree on a naming convention from the outset of your
research project.</li>
</ul>
<div id="exercise-4-typical-folder-organizations" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-4-typical-folder-organizations" class="callout-inner">
<h3 class="callout-title">Exercise 4: Typical folder organizations<a class="anchor" aria-label="anchor" href="#exercise-4-typical-folder-organizations"></a>
</h3>
<div class="callout-content">
<p>(5 min breakout + 7 explanation)</p>
<p>Have a look at the four different folder structures.
<img src=".././fig/07-file_organisation.png" alt="file-organisation-strategies" width="600" class="figure"><em>Figure credits: Ines Boehm</em></p>
<p>The first two: a) b) are recommended for computing, the other two: c)
d) are for more wet/biological projects.</p>
<ul>
<li>Which one is the most similar to your project structure?<br>
</li>
<li>When/why would you use a) and when/why b)</li>
<li>When/why would you use c) and when/why d)</li>
</ul>
</div>
</div>
</div>
<div id="accordionSolution4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution4" aria-expanded="false" aria-controls="collapseSolution4">
  <h4 class="accordion-header" id="headingSolution4">Show me the solution</h4>
</button>
<div id="collapseSolution4" class="accordion-collapse collapse" aria-labelledby="headingSolution4" data-bs-parent="#accordionSolution4">
<div class="accordion-body">
<p>Firstly, the root directory contains a README file that provides an
overview of the project as a whole, a CITATION file that explains how to
reference it and a LICENSE, all three make it
<strong>REUSABLE</strong>.</p>
<p>The <strong>a)</strong> structure is recommended by the <a href="https://doi.org/10.1371/journal.pcbi.1005510" class="external-link">Good enough
practices in scientific computing</a> paper.<br>
This project structure clearly separates the inputs (the raw data) from
the outputs (the results) and the analysis procedure (python code).
Following the same convention (like src folder for code) makes it easy
to find interesting elements, for example the raw data or particular
plotting procedure. Good for sharing analysis project, also for
pipelines where one set of inputs generated the set of outputs in the
step by step manner.</p>
<p>The <strong>b)</strong> structure is called “Organized by analysis”
or “by figure”. As the name suggest it may be recommended to share data
underling a publication. In that way each paper figure is represented by
its raw data, processing scripts and the final results and figure plots.
It is also well suited if each analysis deals with different data type
or different aspect of it. When compared to ‘a)’ it makes easier to
match the required inputs with the computational procedures.</p>
<p>The structure similar to c) is recommended for Brain Imaging Data
Structure <a href="https://doi.org/10.1038/sdata.2016.44" class="external-link">BIDS</a>, as
it is organized by “patient” (in this case patient was replaced by pig
:) ) and type of scans. Here the focus is on individual subject /
samples, for which various data was obtained.</p>
<p>Structured <strong>d)</strong> is useful when we are interested in
outcomes of experimental conditions (here drug treatments). The same set
of samples/subjects/genotypes are exposed to different experimental
variables/conditions and the data are probably compared between all the
samples at the same conditions.</p>
</div>
</div>
</div>
</div>
<p>(4 min teaching) ::::::::::::::::::::::::::::::::::::::::::::::::
spoiler</p>
</section><section id="optional-good-enough-practices-for-scientific-computing-recommendations"><h2 class="section-heading">(Optional) Good enough practices for scientific computing
recommendations<a class="anchor" aria-label="anchor" href="#optional-good-enough-practices-for-scientific-computing-recommendations"></a>
</h2>
<hr class="half-width">
<p>The <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1005510#sec009" class="external-link">Good
enough practices in scientific computing</a> paper makes the following
simple recommendations:</p>
<ul>
<li>Put each project in its own directory, which is named after the
project</li>
<li>Put text documents associated with the project in the ‘doc’
directory</li>
<li>Put raw data and metadata in a ‘data’ directory</li>
<li>Put files generated during cleanup and analysis in a ‘results’
directory</li>
<li>Put project source code in the ‘src’ directory</li>
<li>Put compiled programs in the ‘bin’ directory</li>
<li>Name all files to reflect their content or function:</li>
<li>Use names such as ‘bird_count_table.csv’, ‘notebook.md’, or
‘summarized_results.csv’.</li>
<li>Do not use sequential numbers (e.g., result1.csv, result2.csv) or a
location in a final manuscript (e.g., fig_3_a.png), since those numbers
will almost certainly change as the project evolves.</li>
</ul>
<p>:::::::::::::::::::::::::::::::::::::::::::::::</p>
</section><section id="after-you-have-a-plan"><h2 class="section-heading">After you have a plan<a class="anchor" aria-label="anchor" href="#after-you-have-a-plan"></a>
</h2>
<hr class="half-width">
<p>Your naming conventions might need some adjustments as the project
progresses. Don’t despair, just document it!</p>
<p>If you change the strategy document it in PROJECT_STRUCTURE (or
README) stating why you made the change and when.</p>
<p>Update the locations and names of files which followed the old
convention</p>
</section><section id="backing-up-your-project-files-and-folders"><h2 class="section-heading">Backing up your project files and folders<a class="anchor" aria-label="anchor" href="#backing-up-your-project-files-and-folders"></a>
</h2>
<hr class="half-width">
<ul>
<li>Back up (almost) everything created by a human being or recorded by
a machine as soon as it is created.</li>
<li>Always backup your files in 3 places, at least one should be
off-site.</li>
<li>USB sticks are a failure-prone option and are not a valid solution
for backup of scientific data</li>
<li>A robust backup cannot be achieved manually</li>
</ul>
<p>Do you know how and where to keep 3 copies of your data which are
always up to date?</p>
<p><strong>Secure data preservation is very difficult to achieve without
institutional support and know-how.</strong></p>
<p>One option is cloud storage, but not all data may be put in a public
cloud.</p>
<p>You should always check your institutional guidelines and what
solutions are available in your organisation.</p>
</section><section id="project-file-organization-and-fair-guidelines"><h2 class="section-heading">Project file organization and FAIR guidelines<a class="anchor" aria-label="anchor" href="#project-file-organization-and-fair-guidelines"></a>
</h2>
<hr class="half-width">
<div id="exercise-5-fair-files-3-minutes" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="exercise-5-fair-files-3-minutes" class="callout-inner">
<h3 class="callout-title">Exercise 5: FAIR Files (3 minutes)<a class="anchor" aria-label="anchor" href="#exercise-5-fair-files-3-minutes"></a>
</h3>
<div class="callout-content">
<p>Choose 3 main benefits of a good strategy for folder organisation and
naming conventions.</p>
<ul>
<li>Makes data more findable</li>
<li>Aids in making data more reproducible - projects can be copied
easily</li>
<li>Raw data can be reanalysed multiple times</li>
<li>Naming conventions can be read automatically</li>
<li>Easy to understand content by name, less misunderstandings</li>
<li>Easier to find and share data with others</li>
<li>Easy inspection of the project progress (present files)</li>
<li>Fewer meetings required when sharing data</li>
<li>Time saving</li>
</ul>
<p>Have you realised that the above suggestions mean including valuable
metadata as part of your folder structure and file names?</p>
</div>
</div>
</div>
</section><section id="where-to-next"><h2 class="section-heading">Where to next<a class="anchor" aria-label="anchor" href="#where-to-next"></a>
</h2>
<hr class="half-width">
<p>Bulk renaming of files can be done with the software such as:</p>
<ul>
<li>Ant Renamer</li>
<li>RenameIT</li>
<li>Rename4Mac.</li>
</ul></section><section id="attribution"><h2 class="section-heading">Attribution<a class="anchor" aria-label="anchor" href="#attribution"></a>
</h2>
<hr class="half-width">
<p>Content of this episode was created using the following references as
inspiration:</p>
<ul>
<li><a href="https://doi.org/10.1371/journal.pcbi.1005510" class="external-link">Good enough
practices in scientific computing (Wilson et al., 2017)</a></li>
<li><a href="https://www.data.cam.ac.uk/data-management-guide/organising-your-data" class="external-link">Organising
your data</a></li>
<li><a href="https://www.wur.nl/en/Value-Creation-Cooperation/Collaborating-with-WUR-1/Organising-files-and-folders.htm" class="external-link">Organising
files and folders</a></li>
<li><a href="https://libguides.princeton.edu/c.php?g=102546&amp;p=930626" class="external-link">File
naming</a></li>
<li><a href="https://librarycarpentry.org/lc-fair-research/05-reusable/index.html" class="external-link">Library
Caprenty FAIR Data</a></li>
</ul>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points<a class="anchor" aria-label="anchor" href="#keypoints1"></a>
</h3>
<div class="callout-content">
<ul>
<li>A good file name suggests the file content</li>
<li>Good project organization saves you time</li>
<li>Describe your files organization in PROJECT_STRUCTURE or README
including naming convention</li>
</ul>
</div>
</div>
</div>
<!-- 
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use. 
 -->
</section></section><section id="aio-05-reusable-analysis"><p>Content from <a href="05-reusable-analysis.html">Reusable analysis</a></p>
<hr>
<p>Last updated on 2024-03-22 |
        
        <a href="https://github.com/BioRDM/fair-for-busy-biologists/edit/main/episodes/05-reusable-analysis.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 45 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How keep track of your data analysis procedure?</li>
<li>How to make reproducible plots?</li>
<li>What are Jupyter notebooks</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Recognise benefits of notebooks for analysis</li>
<li>Use a jupyter notebook and modify it</li>
<li>Understand notebooks role in being FAIR</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<hr>
<section id="reusable-analysis"><h2 class="section-heading">Reusable Analysis<a class="anchor" aria-label="anchor" href="#reusable-analysis"></a>
</h2>
<hr class="half-width">
<p>(~50 min teaching)</p>
<p><strong>Data analysis and research are inherently iterative.</strong>
Experimental measurements and their subsequent analysis guide the
research journey. For example, in drug discovery, analyzing initial data
might reveal the need to switch targets if a current one proves
unsuitable. Similarly, with large datasets or omics experiments, initial
exploration often necessitates adjustments to analysis methods and,
potentially, the experimental procedures themselves.</p>
<p>This “<strong>ad-hoc</strong>” analysis, characterized by repeated
exploration, refinement, and adaptation, is crucial for identifying
suitable methods and parameters. Only after multiple iterations does the
process culminate in statistically validated results, ready for
presentation in publications with accompanying graphs. Traditional
methods like pen and paper or spreadsheets struggle to effectively track
this iterative nature of research.</p>
</section><section id="reusable-computing"><h2 class="section-heading">Reusable Computing<a class="anchor" aria-label="anchor" href="#reusable-computing"></a>
</h2>
<hr class="half-width">
<p>We have previously discussed electronic lab notebooks and their
benefit in being FAIR. If you are working with large datasets,
mathematical models, complex visualisations of your data then you might
already be frustrated by having to copy and paste your figures into your
electronic lab notebook after each iteration of new code (or generation
in Excel). You might also have lost track of which code corresponded to
which figure and why you changed your code last time.</p>
<p>There is a simple solution to this: <strong>Computational
notebooks</strong></p>
</section><section id="computational-notebooks---jupyter-notebook-for-fair-practices"><h2 class="section-heading">Computational Notebooks - Jupyter Notebook for FAIR practices<a class="anchor" aria-label="anchor" href="#computational-notebooks---jupyter-notebook-for-fair-practices"></a>
</h2>
<hr class="half-width">
<p>Just like traditional lab notebooks document experiments, <strong><a href="../www.jupyter.org">Jupyter Notebooks</a></strong> serve as digital
workspaces for scientific computing. Instead of physical data like DNA
gels, researchers can embed code, data visualizations (graphs), and
explanations directly within the notebook. This free and popular tool
offers several advantages:</p>
<ul>
<li>Interactive coding: Jupyter Notebooks allow writing and editing code
in various languages like Python, R, or Julia (hence the name “Ju” +
“Py” + “ter”). The results (calculations or graphs) appear instantly
alongside the code.</li>
<li>Integrated Documentation: Code snippets seamlessly blend with
explanations, capturing the entire experimental workflow – from setup
and analysis to results and visualizations.</li>
</ul>
<p><strong>Interactive data exploration</strong>: Data scientists
leverage this format to explore data through an iterative process. They
can write code, view the output, modify it, and repeat, fostering an
interactive loop between researcher and data.<br>
The huge benefit of such notebooks is that source code is mixed with
documentation, thus explaining experimental setup, analysis, results of
tables and plots throughout.</p>
<p>As a basic principle Jupyter Notebooks run on a ‘kernel’ which is
responsible in execution of the code. Generally this ‘kernel’ can run on
your computer or you can use external servers to store and analyse your
data.</p>
<p><strong>Sharing and Collaboration</strong>: Notebooks can be exported
as PDF or HTML files for easy sharing. Additionally, services like <a href="https://nbviewer.org/" class="external-link">nbviewer</a> allow rendering notebooks
directly on GitHub, eliminating the need for software installation.</p>
<p>To show you how easy it is to work with Jupyter Notebooks, we have
created an exercise an exercise where you’ll explore real-world data
from a plant experiment.</p>
<p>We’ll analyze the phenotypes of Arabidopsis plants under short and
long-day light conditions using R programming as an example. We will
create some graphs and test whether there are differences between our
genotypes of arabidopsis.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">SH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode sh" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="ex">The</span> example notebook and data files can be found in instructors folders.  </span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="ex">The</span> most recent version are available via a separated github project:  </span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="ex">[fair-jupyter]</span><span class="er">(</span><span class="ex">https://github.com/BioRDM/fair-jupyter</span><span class="kw">)</span></span></code></pre>
</div>
<div id="exercise-1-basics-of-jupyter-notebooks-5-min" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-1-basics-of-jupyter-notebooks-5-min" class="callout-inner">
<h3 class="callout-title">Exercise 1: basics of Jupyter notebooks (5
min)<a class="anchor" aria-label="anchor" href="#exercise-1-basics-of-jupyter-notebooks-5-min"></a>
</h3>
<div class="callout-content">
<p>Navigate to the jupyter server, we will first show you how to
duplicate a notebook and save it and how to run the code:</p>
<ol style="list-style-type: decimal">
<li><p>Select the notebook titled
‘student_notebook_light_conditions.ipynb’ as depicted below and click
‘Duplicate’. Confirm with Duplicate when you are asked if you are
certain that you want to duplicate the notebook. <img src=".././fig/10-01-duplicate_notebook.png" alt="Figure 1. Duplicate a Jupyter notebook" class="figure"><em>Figure 1. Duplicate
a Jupyter notebook</em></p></li>
<li><p>A copy of the notebook has appeared with the suffix ‘-Copy’ and a
number (<strong>Figure 2a</strong>), select this notebook. Have a look
around the notebook and explore its anatomy (<strong>Figure 2</strong>),
you should see experimental details, an image, and code. If you click on
separate parts of the notebook you can see that it is divided into
individual cells (<strong>Figure 2 e-g</strong>) which are of varying
type (Code, R in this case, or Markdown - <strong>Figure 2d</strong>).
Hashtags are comments within the code and shall help you to interpret
what individual bits of code do.</p></li>
</ol>
<p><img src=".././fig/10-02-jupyter_anatomy.png" alt="Figure 2. Anatomy of a Jupyter notebook: (a) depicts the name of the notebook, (b, c) are toolbars, (c) contains the most commonly used tools, (d) shows of what type - Markdown, Code etc… - the currently selected cell is, and (e-g) are examples of cells, where (e) shows the currently selected cell." class="figure"><br><em>Figure 2. Anatomy of a Jupyter notebook: (a) depicts the name of the
notebook, (b, c) are toolbars, (c) contains the most commonly used
tools, (d) shows of what type - Markdown, Code etc… - the currently
selected cell is, and (e-g) are examples of cells, where (e) shows the
currently selected cell.</em></p>
<ol style="list-style-type: decimal">
<li>Change the title of the notebook (<strong>a</strong>) to your
initials e.g. “student_notebook_light_conditions_IB”</li>
<li>To Save the notebook click on the disk symbol in the toolbar
(<strong>c</strong>).</li>
<li>To run the code select the top cell of the notebook
(<strong>e</strong>) - this is likely pre-selected already - and click
“Run” in the tool bar (<strong>c</strong>). The selected cell jumps one
cell down (<strong>f</strong>).</li>
<li>To avoid having to press Run repeatedly until we are at the end of
the code (try clicking Run two more times and see what happens), we will
show you how to run all the code. In the top tool bar
(<strong>b</strong>) click “Cell” and select “Run All”. (“Cell &gt; Run
All”.)</li>
<li>If you have another look at the notebook you can see that a table,
graph and outputs of statistical testing have been generated.</li>
</ol>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1">Show me the solution</h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<p>If you followed all steps correctly you should have reproduced the
table, a graph and statistical testing. Apart from the pre-filled
markdown text the rendered values of the code should look like this:
<img src=".././fig/10-03-jupyter_dataframe.png" alt="Figure 3. Rendering of data frame" class="figure"><br><em>Figure 3. Rendering of data frame</em> <img src=".././fig/10-04-jupyter_plot.png" alt="Figure 4. Rendering of plot" class="figure"><br><em>Figure 4. Rendering of plot</em></p>
</div>
</div>
</div>
</div>
<div id="exercise-2-how-to-add-and-remove-content-3-min" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="exercise-2-how-to-add-and-remove-content-3-min" class="callout-inner">
<h3 class="callout-title">Exercise 2: how to add and remove content (3
min)<a class="anchor" aria-label="anchor" href="#exercise-2-how-to-add-and-remove-content-3-min"></a>
</h3>
<div class="callout-content">
<p>In your previously saved notebook, we will now show you how to add
text and remove cells within your notebook. Additionally we will show
you how to change code:</p>
<ol style="list-style-type: decimal">
<li>We want to change the author name to your name: Double click on the
cell containing the author name. You can see how the layout of the cell
changes into Markdown, and you can simply change the name.</li>
<li>Press the Run button in the toolbar and the Markdown of the cell
will be rendered again and you can see the text shows as
previously.</li>
<li>We now want to add some details about the “light_results.txt” file
which is being loaded. To add a cell above the code, click the cell
currently above the first lines of code and click the “+” in the
toolbar. This creates a cell below selected cell.</li>
<li>Now add a note about the file which is being loaded and the purpose
of the subsequent analysis: e.g. Loading of results following short- and
long-day light exposure on arabidopsis, followed by visualisation of
differences in chlorophyll/biomas etc… content between genotypes on
short-days and long-days. To show the cell as Markdown and not code,
make sure “Markdown” is selected in the toolbar.</li>
<li>To remove a cell, select it and click on the scissors icon in the
toolbar (This can be undone under Edit &gt; Undo Delete Cells).</li>
<li>To change the output of your graph click on the cell containing the
code below the “Visualise data” title. We want you to change the colours
of the box-plots. You can do this where the comment “# change colour of
groups” is written. Either you can use three different HEX codes (a
6-symbol code representing colours from white to black), or colours
simply by their name - e.g. dark blue, orange…</li>
<li>To save the graph under a different name add your initials to the
file name under which the image is being saved. Press Run again. Your
image should be visible in the overall file hierarchy.</li>
</ol>
</div>
</div>
</div>
<p>(3 min teaching)</p>
</section><section id="jupyter-notebooks-are-easy-to-use-and-great-for-collaborative-efforts"><h2 class="section-heading">Jupyter Notebooks are easy to use and great for collaborative
efforts<a class="anchor" aria-label="anchor" href="#jupyter-notebooks-are-easy-to-use-and-great-for-collaborative-efforts"></a>
</h2>
<hr class="half-width">
<p>Imagine your collaborators have shared this experiment measuring
biomas of <em>Arabidopsis</em> with you and you were able to duplicate
and follow their entire analysis from start to finish, including
interpretation of data. Now you are interested in analysing what biomas
look like on long-days. You are in luck!</p>
<p>Because Jupyter Notebooks are so easy to handle you simply need to
copy and paste the already existing code your collaborators shared and
adapt variables such as short-day to long-day and change the names under
which figures are being saved to avoid duplication.</p>
<div id="exercise-3-add-another-analysis-step-10-min-including-walkthrough" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-3-add-another-analysis-step-10-min-including-walkthrough" class="callout-inner">
<h3 class="callout-title">Exercise 3: add another analysis step (10 min
including walkthrough)<a class="anchor" aria-label="anchor" href="#exercise-3-add-another-analysis-step-10-min-including-walkthrough"></a>
</h3>
<div class="callout-content">
<p>We have shown you how to manipulate text and code in Jupyter
notebooks, now we want you to add data visualisation (a graph) and stats
for long- day light condition: 1. Add additional cells including</p>
<ul>
<li>titles</li>
<li>edited code to depict graph from long-days and not short-days</li>
<li>Figure legend</li>
<li>statistical testing of difference between genotypes on
long-days</li>
<li>interpretation of results of statistical testing.</li>
</ul>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2">Show me the solution</h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" data-bs-parent="#accordionSolution2" aria-labelledby="headingSolution2">
<div class="accordion-body">
<p>## Solution The following code will result in your new graph:</p>
<pre><code><span>    <span class="fu">ggplot</span><span class="op">(</span><span class="fu">subset</span><span class="op">(</span><span class="va">df</span>, <span class="va">light_condition</span> <span class="op">%in%</span> <span class="st">"LD"</span><span class="op">)</span>, </span>
<span>    <span class="co"># subset only SD from &gt;&gt;light condition column for plotting</span></span>
<span>          mapping <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">genotype</span>, y <span class="op">=</span> <span class="va">biomas</span>, fill <span class="op">=</span> <span class="va">genotype</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="co"># x-axis shows genotype, y-axis shows biomas</span></span>
<span>       <span class="fu">geom_boxplot</span><span class="op">(</span>alpha<span class="op">=</span><span class="fl">0.3</span><span class="op">)</span> <span class="op">+</span></span>
<span>       <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Biomas per Genotype on long days"</span>,</span>
<span>           x <span class="op">=</span> <span class="st">"Genotype"</span>, <span class="co"># Title of x-axis</span></span>
<span>           y <span class="op">=</span> <span class="st">"Biomas (g)"</span><span class="op">)</span> <span class="op">+</span> <span class="co"># Title of y-axis</span></span>
<span>       <span class="co"># change colour of groups</span></span>
<span>       <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="fu">c</span><span class="op">(</span><span class="st">"#999999"</span>, <span class="st">"#E69F00"</span>, <span class="st">"#56B4E9"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>       <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>       <span class="fu">theme</span><span class="op">(</span>legend.position<span class="op">=</span><span class="st">"none"</span><span class="op">)</span></span></code></pre>
<p>The following code will result in testing of biomas between genotypes
in long-days - we assign a new variable to separate both analysis.</p>
<pre><code><span>   <span class="va">res.aov.LD</span> <span class="op">&lt;-</span> <span class="fu">aov</span><span class="op">(</span><span class="va">biomas</span> <span class="op">~</span> <span class="va">genotype</span>, data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">df</span>, <span class="va">light_condition</span> <span class="op">%in%</span> <span class="st">"LD"</span><span class="op">)</span><span class="op">)</span></span>
<span>   <span class="co"># Summary of the analysis</span></span>
<span>   <span class="fu">summary</span><span class="op">(</span><span class="va">res.aov.LD</span><span class="op">)</span></span></code></pre>
<p>The following code will result in Tukey multiple pairwise-comparison
testing.</p>
<pre><code><span>     <span class="co"># conduct Tukey multiple pairwise-comparison</span></span>
<span>     <span class="fu">TukeyHSD</span><span class="op">(</span><span class="va">res.aov.LD</span><span class="op">)</span></span></code></pre>
</div>
</div>
</div>
</div>
</section><section id="easy-sharing-of-your-notebook"><h2 class="section-heading">Easy sharing of your notebook<a class="anchor" aria-label="anchor" href="#easy-sharing-of-your-notebook"></a>
</h2>
<hr class="half-width">
<p>We have now managed to not only reproduce code, but we were able to
add to the analysis and interpretation of overall results. To show your
PI and colleagues your results, you want to save the notebook in
readable format.</p>
<div id="exercise-4-sharing-of-your-jupyter-notebook-5-min" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-4-sharing-of-your-jupyter-notebook-5-min" class="callout-inner">
<h3 class="callout-title">Exercise 4: Sharing of your Jupyter Notebook
(5 min)<a class="anchor" aria-label="anchor" href="#exercise-4-sharing-of-your-jupyter-notebook-5-min"></a>
</h3>
<div class="callout-content">
<ol style="list-style-type: decimal">
<li>Download your Notebook (ensure all code has been run) as .html</li>
<li>View the documents and think about why it is important to run all
code before download (try Cell &gt; All Output &gt; Clear and download
your Notebook then and compare)</li>
</ol>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3">Show me the solution</h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" data-bs-parent="#accordionSolution3" aria-labelledby="headingSolution3">
<div class="accordion-body">
<p>It is important all code is run before the notebook is downloaded, as
during download only the text and graphs are saved that are currently
visible in your notebook.</p>
</div>
</div>
</div>
</div>
<p>The example notebook is extremely well documented to show the good
practices when coding. Without such level of comments and explanations
it may not be possible for new-commers to re-use this notebook.</p>
</section><section id="plotting-in-r-or-python"><h2 class="section-heading">Plotting in R or Python<a class="anchor" aria-label="anchor" href="#plotting-in-r-or-python"></a>
</h2>
<hr class="half-width">
<p>Both R and Python offer powerful tools for creating scientific plots,
making them a natural starting point for many programmers. Compared to
Excel, these languages enable the generation of:</p>
<ul>
<li>Professional visualizations: R and Python provide extensive
libraries for creating publication-quality plots with a wider range of
customization options.</li>
<li>Reproducibility: Code-driven plots ensure consistent results and are
easier to share and adapt for specific publication requirements.</li>
<li>Scalability: Scripting allows for the efficient creation of multiple
figures with consistent formatting, saving researchers time and
effort.</li>
</ul>
<p>Key benefits of using code for scientific plots:</p>
<ul>
<li>Customization: Precise control over every plot element (colors,
fonts, axes, etc.) ensures adherence to specific journal formatting
guidelines.</li>
<li>Efficiency: Code facilitates the creation of complex plots or
repetitive tasks, streamlining the workflow.</li>
</ul></section><section id="capturing-the-ad-hoc-analysis-journey"><h2 class="section-heading">Capturing the Ad-hoc Analysis Journey<a class="anchor" aria-label="anchor" href="#capturing-the-ad-hoc-analysis-journey"></a>
</h2>
<hr class="half-width">
<p>Jupyter Notebooks excel at documenting the entire <strong>exploratory
data analysis (EDA)</strong> process. They capture the rationale and
decision-making behind your analysis, serving as a narrative alongside
the code.</p>
<p><strong>Key benefits of Jupyter Notebooks for EDA:</strong></p>
<ul>
<li>
<strong>Comprehensive Documentation:</strong> Markdown cells allow
you to record:
<ul>
<li>Motivations and thought processes leading to the final results.</li>
<li>Details about input data usage.</li>
<li>Specific parameters and functions employed.</li>
<li>Intermediate results and adjustments made throughout the
analysis.</li>
</ul>
</li>
<li>
<strong>Transparency and Reproducibility:</strong> By capturing all
analytical steps, notebooks promote clear communication and ensure the
analysis can be easily replicated.</li>
</ul>
<p><strong>Maintaining Clean and Reusable Notebooks:</strong></p>
<p>While notebooks offer immense advantages, their effectiveness relies
on <strong>adherence to best practices</strong>:</p>
<ul>
<li>
<strong>Meticulous Documentation:</strong> Thoroughly document the
decision-making process and the reasoning behind each step.</li>
<li>
<strong>Data Transparency:</strong> Clearly record details about the
input data, including its source and any cleaning procedures
applied.</li>
<li>
<strong>Parameter Clarity:</strong> Document the chosen parameters
and their significance in the analysis.</li>
<li>
<strong>Data Management:</strong> Strategically decide which data to
retain and why, with clear annotations.</li>
<li>
<strong>Code Clarity:</strong> Employ clear and consistent coding
practices, including meaningful variable and function names, and proper
code formatting.</li>
<li>
<strong>Self-Contained Notebooks:</strong> Ensure your notebook is
shipped with all necessary file inputs and a comprehensive description
of the runtime environment.</li>
</ul>
<p><strong>Strengths of Jupyter Notebooks:</strong></p>
<ul>
<li>
<strong>Orchestrating Analyses:</strong> Notebooks efficiently
manage step-by-step operations in R, Python, or even shell scripting
(supporting all three languages simultaneously).</li>
<li>
<strong>Parameter Management:</strong> They effectively capture and
manage analysis parameters.</li>
<li>
<strong>Interpretation and Insights:</strong> Notebooks facilitate
the inclusion of interpretations and insights alongside the code,
enriching the overall analysis.</li>
<li>
<strong>Interactive Interface:</strong> They can function as a
dynamic user interface, allowing users to modify parameters directly
within the notebook based on the provided instructions.</li>
</ul>
<p><strong>Limitations to Consider:</strong></p>
<ul>
<li>
<strong>Not a Replacement for IDEs:</strong> Jupyter Notebooks are
not intended to fully replace Integrated Development Environments
(IDEs). Complex, modularized code should be written in separate
packages.</li>
<li>
<strong>Unsuitable for Long Programs:</strong> While notebooks can
handle smaller scripts, lengthy executable code is better suited for
compilation into standalone modules.</li>
</ul>
<div id="exercise-5-accessibility-of-jupyter-notebooks-5-min" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="exercise-5-accessibility-of-jupyter-notebooks-5-min" class="callout-inner">
<h3 class="callout-title">Exercise 5: Accessibility of Jupyter Notebooks
(5 min)<a class="anchor" aria-label="anchor" href="#exercise-5-accessibility-of-jupyter-notebooks-5-min"></a>
</h3>
<div class="callout-content">
<p>On a scale from -2 to 2, how do you feel about the following
statements (R is interchangeable with Python), where -2 (strongly
disagree), 0 no pinion to +2 strongly agree:</p>
<ul>
<li>making graphs for a subset of data is easier in R than in Excel</li>
<li>it is easier to filter for data in R than in Excel</li>
<li>it is easier to apply formulas or calculations in R than in
Excel</li>
<li>it is easier to generate a series of plots with similar layout in R
than Excel</li>
<li>it is easier to do large scale data processing in R than in
Excel</li>
<li>using notebooks does not require any programming knowledge</li>
<li>notebooks give you a better overview of your data processing than
Excel</li>
<li>Jupyter is free, whilst a Microsoft Office (+Excel) suite costs
$149.99, this alone is an incentive to use Jupyter notebooks as not all
individuals have Microsoft Office supported by their employers</li>
<li>you need to learn R to do any data processing in notebook</li>
</ul>
</div>
</div>
</div>
</section><section id="computing-in-r-and-python"><h2 class="section-heading">Computing in R and Python<a class="anchor" aria-label="anchor" href="#computing-in-r-and-python"></a>
</h2>
<hr class="half-width">
<p>Here’s a revised version of the text about using R and Python for
scientific computing:</p>
<p><strong>Starting Strong in R and Python</strong></p>
<p>While both R and Python are lauded for their flexibility and ease of
use, this very characteristic can lead to the development of poor coding
habits. Their accessible nature can tempt users to prioritize quick
results over well-structured, maintainable code.</p>
<p>To ensure a smooth journey into scientific computing, we recommend a
phased approach:</p>
<ol style="list-style-type: decimal">
<li>
<strong>Grasp the Fundamentals:</strong> Begin by mastering the
creation of basic plots, data cleaning techniques, and data table
manipulation.</li>
<li>
<strong>Solidify Software Engineering Principles:</strong> Next,
delve into the core concepts of software engineering and establish best
practices for writing clean and maintainable code.</li>
<li>
<strong>Progress to Advanced Analysis:</strong> Once a solid
foundation is built, gradually progress to tackling more complex
analytical tasks, constructing data processing pipelines, and crafting
comprehensive workflows.</li>
</ol>
<p><strong>Addressing Environment Inconsistencies:</strong></p>
<p>A significant challenge with R and Python is their dependence on the
specific libraries installed on a user’s machine. Inconsistent library
versions or even the presence of system-specific libraries can lead to
code that functions perfectly on one machine but fails on another.</p>
<p><strong>Mitigating Versioning Issues:</strong></p>
<p>Fortunately, solutions exist to address these environment
inconsistencies. Here are two notable examples:</p>
<ul>
<li>
<strong>Conda (Python):</strong> This package management system
simplifies installation, updates, and dependency management for Python
packages. It also facilitates the creation of isolated project
environments, ensuring each project utilizes the correct set of
libraries and their specific versions. While Conda supports various
languages, it is particularly well-suited for Python environments.</li>
<li>
<strong>renv (R):</strong> In the realm of R, the <code>renv</code>
package offers a robust alternative for managing project
dependencies.</li>
</ul>
<p><strong>Maintaining Transparency:</strong></p>
<p>Regardless of the chosen package management system, it’s crucial to
meticulously track the libraries used during code development, including
their specific versions. This practice fosters transparency,
reproducibility, and simplifies collaboration by ensuring everyone
involved has access to the exact software dependencies required to run
the code successfully.</p>
<p>This revised version emphasizes the importance of adopting good
coding practices from the outset. It outlines a recommended learning
path and highlights the challenges associated with environment
inconsistencies. Additionally, it introduces solutions for managing
package dependencies effectively.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">SH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode sh" tabindex="0"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="co">## Attribution</span></span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a> <span class="ex">Content</span> of this episode was adopted after</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a> <span class="ex">*</span> [Why Jupyter is a data scientists<span class="st">' computational notebook of choice](https://doi.org/10.1038/d41586-018-07196-1)</span></span></code></pre>
</div>
<div id="further-reading" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="further-reading" class="callout-inner">
<h3 class="callout-title">Further Reading<a class="anchor" aria-label="anchor" href="#further-reading"></a>
</h3>
<div class="callout-content">
<ul>
<li><a href="https://reproducible-analysis-workshop.readthedocs.io/en/latest/4.Jupyter-Notebook.html" class="external-link">Reproducible
analysis and Research Transparency</a></li>
<li><a href="https://www.browserstack.com/guide/coding-standards-best-practices" class="external-link">Coding
Standards best practices</a></li>
</ul>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points<a class="anchor" aria-label="anchor" href="#keypoints1"></a>
</h3>
<div class="callout-content">
<ul>
<li>Jupyter Notebooks are useful tools to share analysis with
non-programmers</li>
<li>One single document can visualise background, results, formulae/code
and metadata</li>
<li>One single document helps to make your work more understandable,
repeatable and shareable</li>
</ul>
</div>
</div>
</div>
<!-- 
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use. 
 -->
</section></section><section id="aio-06-data-repositories"><p>Content from <a href="06-data-repositories.html">Public repositories</a></p>
<hr>
<p>Last updated on 2024-03-22 |
        
        <a href="https://github.com/BioRDM/fair-for-busy-biologists/edit/main/episodes/06-data-repositories.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 40 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>Where can I deposit datasets?</li>
<li>What are general data repositories?</li>
<li>How to find a repository?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>See the benefits of using research data repositories.</li>
<li>Differentiate between general and specific repositories.</li>
<li>Find a suitable repository.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section id="what-are-research-data-repositories-4-min-teaching"><h2 class="section-heading">What are research data repositories? (4 min teaching)<a class="anchor" aria-label="anchor" href="#what-are-research-data-repositories-4-min-teaching"></a>
</h2>
<hr class="half-width">
<p>Research data repositories are online repositories that enable the
preservation, curation and publication of research ‘products’. These
repositories are mainly used to deposit research ‘data’. However, the
scope of the repositories is broader as we can also deposit/publish
‘code’ or ‘protocols’ (as we saw with protocols.io).</p>
<p>There are general “data agnostic” repositories, for example:</p>
<ul>
<li>
<a href="http://datadryad.org" class="external-link">Dryad</a>,</li>
<li>
<a href="http://zenodo.org" class="external-link">Zenodo</a>,</li>
<li>
<a href="http://figshare.com" class="external-link">FigShare</a>,</li>
<li>
<a href="http://thedata.org" class="external-link">Dataverse</a>.</li>
</ul>
<p>Or domain specific, for example:</p>
<ul>
<li>
<a href="https://www.uniprot.org/" class="external-link">UniProt</a> protein data,</li>
<li>
<a href="https://www.ncbi.nlm.nih.gov/genbank/" class="external-link">GenBank</a> sequence
data,</li>
<li>
<a href="https://www.ebi.ac.uk/metabolights/" class="external-link">MetaboLights</a>
metabolomics data</li>
<li>
<a href="https://github.com/" class="external-link">GitHub</a> for code.</li>
</ul>
<p>Research outputs should be submitted to discipline/domain-specific
repositories whenever it is possible. When such a resource does not
exist, data should be submitted to a ‘general’ repository. Research data
repositories are a key resource to help in data FAIRification as they
assure Findability and Accessibility.</p>
<div id="exercise-1-public-general-record-5-min" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="exercise-1-public-general-record-5-min" class="callout-inner">
<h3 class="callout-title">Exercise 1: Public general record (5 min)<a class="anchor" aria-label="anchor" href="#exercise-1-public-general-record-5-min"></a>
</h3>
<div class="callout-content">
<p>Have a look at the following record for data set in Zenodo
repository: <a href="https://doi.org/10.5281/zenodo.5045374" class="external-link"><img src="https://zenodo.org/badge/DOI/10.5281/zenodo.5045374.svg" alt="DOI" class="figure"></a></p>
<p>We have discussed which elements of a Zenodo record makes it
FAIR.</p>
<p>Now, skim through the data set description (HINT there is also a
README), try to judge the following, and indicate your evaluation using
marks from 0 to 5 (5 best) as to whether:</p>
<ul>
<li>It is clear what the content of the data set is:</li>
<li>It is clear why the data could be used (i.e., what for):</li>
<li>It is well described:</li>
<li>How confident will you be to work with this data set:</li>
<li>How easy it is to access the data set content:</li>
<li>Your team datasets are equally well described (or better):</li>
</ul>
</div>
</div>
</div>
<p>(4 min discussion)</p>
<div id="exercise-2-datasets-discovery-5-min" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-2-datasets-discovery-5-min" class="callout-inner">
<h3 class="callout-title">Exercise 2: Datasets discovery (5 min)<a class="anchor" aria-label="anchor" href="#exercise-2-datasets-discovery-5-min"></a>
</h3>
<div class="callout-content">
<p>Try to find: - data sets related to neuromuscular junction in Zenodo
Judge the following, indicating your assessment using marks from 0 to 5
(5 best)</p>
<ul>
<li>how easy it is to find similar or interesting data sets:</li>
<li>It is clear what the content of the other data sets are:</li>
<li>It is clear why the data could be used (ie what for):</li>
<li>They are well described:</li>
</ul>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1">Show me the solution</h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<p>Zenodo is a good place to keep your data separate from paper. It
gives access to all files, allowing you to cite the data as well (or
instead of) the paper.</p>
<p>However, it is not to good for discovery and does not enforce most
metadata!</p>
</div>
</div>
</div>
</div>
<p>(3 min teaching)</p>
<div id="minimal-data-set" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="minimal-data-set" class="callout-inner">
<h3 class="callout-title">Minimal data set<a class="anchor" aria-label="anchor" href="#minimal-data-set"></a>
</h3>
<div class="callout-content">
<p>Minimal data set to consist of the data required to replicate all
study findings reported in the article, as well as related metadata and
methods.</p>
<ul>
<li>The values behind the means, standard deviations and other measures
reported;</li>
<li>The values used to build graphs;</li>
<li>The points extracted from images for analysis.</li>
</ul>
<p>(no need for raw data if the standard in the field is to share data
that have been processed)</p>
<p><a href="https://journals.plos.org/plosbiology/s/data-availability" class="external-link">PLOS</a></p>
</div>
</div>
</div>
<div id="exercise-3.-domain-specific-repositories" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-3.-domain-specific-repositories" class="callout-inner">
<h3 class="callout-title">Exercise 3. Domain specific repositories<a class="anchor" aria-label="anchor" href="#exercise-3.-domain-specific-repositories"></a>
</h3>
<div class="callout-content">
<p>(5 min + 5).</p>
<p>Select one of the following repositories based on your
expertise/interests:</p>
<ul>
<li>Have a look at mRNAseq accession ‘E-MTAB-7933’ in <a href="https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-7933/" class="external-link">ArrayExpress</a><br>
</li>
<li>Have a look at microscopy ‘project-1101’ in <a href="https://idr.openmicroscopy.org/webclient/?show=project-1101" class="external-link">IDR</a><br>
</li>
<li>Have a look at the synthethic part record
‘SubtilinReceiver_spaRK_separated’ within the ‘bsu’ collection in <a href="https://synbiohub.org/public/bsu/SubtilinReceiver_spaRK_separated/1" class="external-link">SynBioHub</a><br>
</li>
<li>Have a look at the proteomics record ‘PXD013039’ in <a href="https://www.ebi.ac.uk/pride/archive/projects/PXD013039" class="external-link">PRIDE</a><br>
</li>
<li>Have a look at the metabolomics record ‘MTBLS2289’ in <a href="https://www.ebi.ac.uk/metabolights/MTBLS2289/descriptors" class="external-link">Metabolights</a><br>
</li>
<li>Have a look at the scripts deposit ‘RNA-Seq-validation’ in <a href="https://github.com/PauloFlores/RNA-Seq-validation" class="external-link">GitHub</a>
</li>
</ul>
<p>Report to the group, what advantages can you see in using a specific
repository over a generalist repository like Zenodo.</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2">Show me the solution</h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" data-bs-parent="#accordionSolution2" aria-labelledby="headingSolution2">
<div class="accordion-body">
<p>Some advantages are:</p>
<ul>
<li>The repository is more relevant to your discipline than a generalist
one.</li>
<li>Higher exposure (people looking for those specific types of data
will usually first look at the specific repository).</li>
<li>Higher number of citations (see above).</li>
</ul>
</div>
</div>
</div>
</div>
</section><section id="how-do-we-choose-a-research-data-repository"><h2 class="section-heading">How do we choose a research data repository?<a class="anchor" aria-label="anchor" href="#how-do-we-choose-a-research-data-repository"></a>
</h2>
<hr class="half-width">
<p>(3 min teaching) As a general rule, your research needs to be
deposited in discipline/data specific repository. If no specific
repository can be found, then you can use a generalist repository.
Having said this, there are tons of data repositories to choose from.
Choosing one can be time consuming and challenging as well. So how do
you go about finding a repository:</p>
<ul>
<li>Check the publisher’s / funder’ recommended list of repositories,
some of which can be found below:
<ul>
<li><a href="https://www.springernature.com/gp/authors/research-data-policy/recommended-repositories" class="external-link">BioMed
Central / Springer Nature</a></li>
<li><a href="https://submit.elifesciences.org/html/elife_author_instructions.html#policies" class="external-link">eLife</a></li>
<li><a href="https://www.elsevier.com/about/policies/research-data" class="external-link">Elsevier</a></li>
<li><a href="https://www.embopress.org/page/journal/14602075/authorguide#datadeposition" class="external-link">EMBO
Press</a></li>
<li><a href="https://f1000research.com/for-authors/data-guidelines" class="external-link">F1000
Research</a></li>
<li><a href="https://academic.oup.com/gigascience/pages/instructions_to_authors" class="external-link">GIGAscience
- OUP</a></li>
<li><a href="https://journals.plos.org/plosbiology/s/recommended-repositories" class="external-link">PLoS</a></li>
<li><a href="https://www.nature.com/sdata/policies/repositories" class="external-link">Scientific
Data - Nature</a></li>
<li><a href="https://authorservices.taylorandfrancis.com/data-sharing-policies/repositories/" class="external-link">Taylor
and Francis</a></li>
<li><a href="https://bbsrc.ukri.org/research/resources/" class="external-link">BBSRC</a></li>
<li><a href="https://nerc.ukri.org/research/sites/environmental-data-service-eds/policy/" class="external-link">NERC</a></li>
<li><a href="https://royalsociety.org/journals/ethics-policies/data-sharing-mining/" class="external-link">Royal
Society</a></li>
<li><a href="https://wellcomeopenresearch.org/for-authors/data-guidelines" class="external-link">Wellcome
Open Research</a></li>
</ul>
</li>
<li>Check <a href="https://fairsharing.org/recommendations/?q=" class="external-link">Fairsharing
recommendations</a>
</li>
<li>or check the <a href="https://www.re3data.org/" class="external-link">Registry of research
data repositories - re3data</a>
</li>
</ul>
<div id="exercise-4-finding-a-repository-5-min-2-min-discussion." class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-4-finding-a-repository-5-min-2-min-discussion." class="callout-inner">
<h3 class="callout-title">Exercise 4: Finding a repository (5 min + 2
min discussion).<a class="anchor" aria-label="anchor" href="#exercise-4-finding-a-repository-5-min-2-min-discussion."></a>
</h3>
<div class="callout-content">
<ol style="list-style-type: lower-alpha">
<li>Using Fairsharing (<a href="https://fairsharing.org/" class="external-link uri">https://fairsharing.org/</a>) find a repo for flow
cytometry</li>
<li>once done, search for repository for genomics data.</li>
</ol>
<p>Note to instructor: Fairsharing gives few options, people may give
different answer follow up why they selected particular ones.</p>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3">Show me the solution</h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" data-bs-parent="#accordionSolution3" aria-labelledby="headingSolution3">
<div class="accordion-body">
<ol style="list-style-type: lower-alpha">
<li>FlowRepository<br>
</li>
<li>GEO/SRA and ENA/ArrayExpress are good examples. Interestingly these
repositories do not issue a DOI.</li>
</ol>
</div>
</div>
</div>
</div>
<p>(6 min teaching)</p>
<p>A list of UoE BioRDM’s recommended data repositories can be found <a href="https://www.wiki.ed.ac.uk/display/RDMS/Suggested+data+repositories" class="external-link">here</a>.</p>
</section><section id="what-comes-first-the-repository-or-the-metadata"><h2 class="section-heading">What comes first? the repository or the metadata?<a class="anchor" aria-label="anchor" href="#what-comes-first-the-repository-or-the-metadata"></a>
</h2>
<hr class="half-width">
<p>Finding a repository first may help in deciding what metadata to
collect and how!</p>
</section><section id="extra-features"><h2 class="section-heading">Extra features<a class="anchor" aria-label="anchor" href="#extra-features"></a>
</h2>
<hr class="half-width">
<p>It is also worth considering that some repositories offer extra
features, such as running simulations or providing visualisation. For
example, <a href="https://fairdomhub.org/" class="external-link">FAIRDOMhub</a> can run model
simulations and has project structures. Do not forget to take this into
account when choosing your repository. Extra features might come in
handy.</p>
</section><section id="can-github-be-cited"><h2 class="section-heading">Can GitHub be cited?<a class="anchor" aria-label="anchor" href="#can-github-be-cited"></a>
</h2>
<hr class="half-width">
<p>As github is a tool for version controll and life code, it may not
alway match the version of the code which was described in a
publication.</p>
<p>To solve this issue you should create a snapshot of your repository
and deposit it in a general repository to obtain a DOI.</p>
<p>For example, Zenodo can automatically integrate with github and
create a new deposit for each of your releases.</p>
</section><section id="evaluating-a-research-data-repository"><h2 class="section-heading">Evaluating a research data repository<a class="anchor" aria-label="anchor" href="#evaluating-a-research-data-repository"></a>
</h2>
<hr class="half-width">
<p>You can evaluate the repositories by following this criteria:</p>
<ul>
<li>who is behind it, what is its funding</li>
<li>quality of interaction: is the interaction for purposes of data
deposit or reuse efficient, effective and satisfactory for you?</li>
<li>take-up and impact: what can I put in it? Is anyone else using it?
Will others be able to find stuff deposited in it? Is the repository
linked to other data repositories so I don’t have to search tehre as
well? Can anyone reuse the data? Can others cite the data, and will
depositing boost citations to related papers?<br>
</li>
<li>policy and process: does it help you meet community standards of
good practice and comply with policies stipulating data deposit?</li>
</ul>
<p>An interesting take can be found at Peter Murray-Rust’s blog post <a href="https://blogs.ch.cam.ac.uk/pmr/2011/08/19/criteria-for-successful-repositories/" class="external-link">Criteria
for succesful repositories</a>.</p>
<div id="exercise-5-wrap-up-discussion-5-min." class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="exercise-5-wrap-up-discussion-5-min." class="callout-inner">
<h3 class="callout-title">Exercise 5: Wrap up discussion (5 min).<a class="anchor" aria-label="anchor" href="#exercise-5-wrap-up-discussion-5-min."></a>
</h3>
<div class="callout-content">
<p>Discuss the following questions:</p>
<ul>
<li>Why is choosing a domain specific repositories over zenodo more
FAIR?</li>
<li>How can selecting a repository for your data as soon as you do an
experiment (or even before!) benefit your research and help your data
become FAIR?</li>
<li>What is your favourite research data repository? Why?</li>
</ul>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">SH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode sh" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="co">## Attribution</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="ex">Content</span> of this episode was adapted or inspired by:.</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="ex">-</span> [FAIR principles]<span class="er">(</span><span class="ex">https://www.go-fair.org/fair-principles/</span><span class="kw">)</span></span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="ex">-</span> [BioRDM suggested data repositories]<span class="er">(</span><span class="ex">https://www.wiki.ed.ac.uk/display/RDMS/Suggested+data+repositories</span><span class="kw">)</span></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="ex">-</span> [DCC <span class="at">-</span> How can we evaluate data repositories<span class="pp">?</span>]<span class="er">(</span><span class="ex">https://www.dcc.ac.uk/news/how-can-we-evaluate-data-repositories-pointers-dryaduk</span><span class="kw">)</span></span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a><span class="ex">-</span> [Criteria for succesful repositories]<span class="er">(</span><span class="ex">https://blogs.ch.cam.ac.uk/pmr/2011/08/19/criteria-for-successful-repositories/</span><span class="kw">)</span></span></code></pre>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points<a class="anchor" aria-label="anchor" href="#keypoints1"></a>
</h3>
<div class="callout-content">
<ul>
<li>Repositories are the main means for sharing research data.</li>
<li>You should use data-type specific repository whenever possible.</li>
<li>Repositories are the key players in data reuse.</li>
</ul>
</div>
</div>
</div>
<!-- 
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use. 
 -->
</section></section><section id="aio-07-journey-to-FAIR"><p>Content from <a href="07-journey-to-FAIR.html">Journey to be FAIR</a></p>
<hr>
<p>Last updated on 2024-03-22 |
        
        <a href="https://github.com/BioRDM/fair-for-busy-biologists/edit/main/episodes/07-journey-to-FAIR.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 20 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What will your journey to be FAIR productive entail?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Discuss steps and changes in your habits you will take after this
course</li>
<li>List of resources that will help you achieve these goals</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section id="better-research-by-better-sharing"><h2 class="section-heading">Better research by better sharing<a class="anchor" aria-label="anchor" href="#better-research-by-better-sharing"></a>
</h2>
<hr class="half-width">
<p>For many of us, data management or output sharing in general are
considered a burden rather than a useful activity. Part of the problem
is our bad timing and lack of planning.</p>
<p><strong>Data management is a continuous process</strong></p>
<p><img src=".././fig/01-sharing_as_part_of_workflow.jpg" alt="Figure 5.2. Sharing as part of the workflow" class="figure"><em>Figure credits:
Tomasz Zielinski and Andrés Romanowski</em></p>
<p>When should you engage in data sharing and open practices?</p>
<ul>
<li>Data management should be done throughout the duration of your
project.</li>
<li>If you wait till the end, it will take a massive effort on your side
and will be more of a burden than a benefit.</li>
<li>The overheads of making data suitable for sharing at each stage of
data life-cycle are small comparing to the other actions involved</li>
<li>Taking the time to do effective data management will help you
understand your data better and make it easier to find when you need it
(for example when you need to write a manuscript or a thesis!).</li>
<li>All the practices that enable others to access and use your outcomes
directly benefit you and your group</li>
</ul>
<div id="challenge1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Challenge<a class="anchor" aria-label="anchor" href="#challenge1"></a>
</h3>
<div class="callout-content">
<p>Exercise 1. Knowledge gap</p>
<p>Read through the following activities / practices, type next to each
-1 if you do not perform it ? if you are not completely sure what it
stands for 0 if you only learnt about it at this workshop +1 if you
adheres / practices it</p>
<ul>
<li>Include a license with datasets.</li>
<li>Include a license with code/scripts.</li>
<li>Use Git for version control.</li>
<li>Create DOIs for datasets and code.</li>
<li>Add a “Date Availability” section to a manuscript.</li>
<li>Use minimal information standards.</li>
<li>Use a generic data repository.</li>
<li>Use a domain-specific data repository.</li>
<li>Develop description templates for various lab techniques.</li>
<li>Store data on a shared network drive.</li>
<li>Implement an automatic backup solution for files.</li>
<li>Follow a consistent file naming convention.</li>
<li>Create a standard project folder structure.</li>
<li>Use electronic lab notebooks.</li>
<li>Create figures and plots in Python or R.</li>
<li>Select a data repository.</li>
<li>Be familiar with non-restrictive licenses.</li>
<li>Create a README file for each dataset.</li>
<li>Utilize controlled vocabularies.</li>
<li>Obtain an ORCID identifier.</li>
<li>Maintain a dedicated folder or database for protocols and SOPs.</li>
<li>Establish a system for referencing different versions of
protocols.</li>
<li>Adhere to conventions for tidy data tables.**</li>
<li>Use Jupyter Notebooks or R Markdown.</li>
<li>Include PIDs (e.g., UniProt, GenBank) in the data description.</li>
<li>Use a database for biosamples, strains, etc.</li>
<li>Grant yourself access to all group data from your personal computer
(optional).</li>
<li>Utilize the data management tools and resources offered by your
organization.</li>
<li>Seek support from your organization for data management.</li>
</ul>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1">Show me the solution</h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<ul>
<li>Try to find out about all the actions you were not familiar
with</li>
<li>revisit your -1 practices and decide if not to change them to 1</li>
</ul>
</div>
</div>
</div>
</div>
<div id="exercise-1-your-journey-to-be-fairproductive" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-1-your-journey-to-be-fairproductive" class="callout-inner">
<h3 class="callout-title">Exercise 1: Your journey to be
FAIRproductive<a class="anchor" aria-label="anchor" href="#exercise-1-your-journey-to-be-fairproductive"></a>
</h3>
<div class="callout-content">
<p>What are the things you are going to change in your work habits, or
actions you are going to take after this course?</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2">Show me the solution</h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<p>Some good actions to take:</p>
<ul>
<li>For each paper I publish I will make a Zenodo deposit with all the
data files</li>
<li>I will use GitHub for all my code instead of using folders</li>
<li>I will start using ELNs</li>
<li>Write a data management plan for my project</li>
<li>Reorganise my folder structure</li>
<li>Use templates</li>
<li>Start using readme files (make a readme template)</li>
<li>Write protocols on protocols.io</li>
<li>Back up my data accordingly</li>
<li>Think as a user of my data instead of just a creator</li>
<li>Start using Jupyter notebooks, or RMarkdown to describe analysis
workflows</li>
<li>Write intelligible metadata</li>
</ul>
</div>
</div>
</div>
</div>
<div id="resources" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="resources" class="callout-inner">
<h3 class="callout-title">Resources<a class="anchor" aria-label="anchor" href="#resources"></a>
</h3>
<div class="callout-content">
<p>The full version of this course: <a href="https://carpentries-incubator.github.io/fair-bio-practice/" class="external-link">FAIR
in (bio) practice</a></p>
<p>The BioRDM team has a lot of materials on their <a href="https://www.wiki.ed.ac.uk/display/RDMS/" class="external-link">BioRDM wiki</a>.</p>
<p>Practical recipes and interactive guides:</p>
<ul>
<li><a href="https://faircookbook.elixir-europe.org/" class="external-link uri">https://faircookbook.elixir-europe.org/</a></li>
<li><a href="https://rdmkit.elixir-europe.org/" class="external-link uri">https://rdmkit.elixir-europe.org/</a></li>
</ul>
<p>Finding resources for example repositories or standards</p>
<ul>
<li><a href="https://fairsharing.org/" class="external-link uri">https://fairsharing.org/</a></li>
<li><a href="https://www.re3data.org/" class="external-link uri">https://www.re3data.org/</a></li>
<li>BioRDM wiki</li>
</ul>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points<a class="anchor" aria-label="anchor" href="#keypoints1"></a>
</h3>
<div class="callout-content">
<ul>
<li>There are simple steps to help make your data more FAIR throughout
the research data lifecycle</li>
<li>Implement these steps throughout to keep track of your data and
changes</li>
</ul>
</div>
</div>
</div>
<!-- 
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use. 
 -->
</section></section>
</div>
    </main>
</div>
<!-- END  : inst/pkgdown/templates/content-extra.html -->

      </div>
<!--/div.row-->
      		<footer class="row footer mx-md-3"><hr>
<div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>
        
        <a href="https://github.com/BioRDM/fair-for-busy-biologists/edit/main/README.md" class="external-link">Edit on GitHub</a>
        
	
        | <a href="https://github.com/BioRDM/fair-for-busy-biologists/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://github.com/BioRDM/fair-for-busy-biologists/" class="external-link">Source</a></p>
				<p><a href="https://github.com/BioRDM/fair-for-busy-biologists/blob/main/CITATION" class="external-link">Cite</a> | <a href="mailto:bio_rdm@ed.ac.uk">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">
        
        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>
        
        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.16.3" class="external-link">sandpaper (0.16.3)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.4" class="external-link">pegboard (0.7.4)</a>, and <a href="https://github.com/carpentries/varnish/tree/1.0.1" class="external-link">varnish (1.0.1)</a></p>
			</div>
		</footer>
</div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "TrainingMaterial",
  "@id": "https://BioRDM.github.io/fair-for-busy-biologists/instructor/aio.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "FAIR, data, data-management, biomedical",
  "name": "All in One View",
  "creativeWorkStatus": "active",
  "url": "https://BioRDM.github.io/fair-for-busy-biologists/instructor/aio.html",
  "identifier": "https://BioRDM.github.io/fair-for-busy-biologists/instructor/aio.html",
  "dateCreated": "2024-03-22",
  "dateModified": "2024-03-26",
  "datePublished": "2024-03-26"
}

  </script><script>
		feather.replace();
	</script><!-- Matomo
    2022-11-07: we have gotten a notification that we have an overage for our
    tracking and I'm pretty sure this has to do with Workbench usage.
    Considering that I am not _currently_ using this tracking because I do not
    yet know how to access the data, I am turning this off for now.
  <script>
    var _paq = window._paq = window._paq || [];
    /* tracker methods like "setCustomDimension" should be called before "trackPageView" */
    _paq.push(["setDocumentTitle", document.domain + "/" + document.title]);
    _paq.push(["setDomains", ["*.preview.carpentries.org","*.datacarpentry.github.io","*.datacarpentry.org","*.librarycarpentry.github.io","*.librarycarpentry.org","*.swcarpentry.github.io", "*.carpentries.github.io"]]);
    _paq.push(["setDoNotTrack", true]);
    _paq.push(["disableCookies"]);
    _paq.push(['trackPageView']);
    _paq.push(['enableLinkTracking']);
    (function() {
          var u="https://carpentries.matomo.cloud/";
          _paq.push(['setTrackerUrl', u+'matomo.php']);
          _paq.push(['setSiteId', '1']);
          var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
          g.async=true; g.src='https://cdn.matomo.cloud/carpentries.matomo.cloud/matomo.js'; s.parentNode.insertBefore(g,s);
        })();
  </script>
  End Matomo Code -->
</body>
</html><!-- END:   inst/pkgdown/templates/layout.html-->

